diff -wpruN --no-dereference '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -254,17 +254,6 @@ AM_CONDITIONAL([HAVE_PYTEST], [test -n "
 #
 AC_PATH_PROG([XSLTPROC], [xsltproc])
 
-#
-# Using Solaris linker with gcc on Solaris breaks Thread Local Storage
-#
-AS_CASE([$host],
-	[*-solaris*],[
-	    AS_IF([test "$GCC" = "yes"],
-		  [LDFLAGS="$LDFLAGS -zrelax=transtls"
-		   AC_MSG_WARN([When using GNU C Compiler on Solaris, -zrelax=transtls linker flag is used to fix bug in Thread Local Storage])
-		  ])
-	])
-
 AC_CHECK_HEADERS([sys/param.h sys/socket.h])
 AC_CHECK_HEADERS([fcntl.h regex.h sys/time.h unistd.h sys/mman.h sys/sockio.h sys/select.h sys/sysctl.h net/if6.h net/route.h linux/netlink.h linux/rtnetlink.h], [], [],
 		 [$ac_includes_default
