diff -wpruN '--exclude=*.orig' a~/bin/tests/test_server.c a/bin/tests/test_server.c
--- a~/bin/tests/test_server.c	1970-01-01 00:00:00
+++ a/bin/tests/test_server.c	1970-01-01 00:00:00
@@ -202,7 +202,7 @@ teardown(void) {
 }
 
 static void
-yield(void) {
+_yield(void) {
 	sigset_t sset;
 	int sig;
 
@@ -306,7 +306,7 @@ run(void) {
 	}
 	REQUIRE(result == ISC_R_SUCCESS);
 
-	yield();
+	_yield();
 
 	isc_nm_stoplistening(sock);
 	isc_nmsocket_close(&sock);
