diff -wpruN '--exclude=*.orig' a~/src/node_crypto.cc a/src/node_crypto.cc
--- a~/src/node_crypto.cc	1970-01-01 00:00:00
+++ a/src/node_crypto.cc	1970-01-01 00:00:00
@@ -4582,11 +4582,19 @@ static unsigned int GetBytesOfRS(const M
   int bits, base_id = EVP_PKEY_base_id(pkey.get());
 
   if (base_id == EVP_PKEY_DSA) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    const DSA* dsa_key = EVP_PKEY_get0_DSA(pkey.get());
+#else
     DSA* dsa_key = EVP_PKEY_get0_DSA(pkey.get());
+#endif
     // Both r and s are computed mod q, so their width is limited by that of q.
     bits = BN_num_bits(DSA_get0_q(dsa_key));
   } else if (base_id == EVP_PKEY_EC) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    const EC_KEY* ec_key = EVP_PKEY_get0_EC_KEY(pkey.get());
+#else
     EC_KEY* ec_key = EVP_PKEY_get0_EC_KEY(pkey.get());
+#endif
     const EC_GROUP* ec_group = EC_KEY_get0_group(ec_key);
     bits = EC_GROUP_order_bits(ec_group);
   } else {
