diff -wpruN '--exclude=*.orig' a~/tools/genv8constants.py a/tools/genv8constants.py
--- a~/tools/genv8constants.py	1970-01-01 00:00:00
+++ a/tools/genv8constants.py	1970-01-01 00:00:00
@@ -20,7 +20,7 @@ if len(sys.argv) != 3:
 outfile = open(sys.argv[1], 'w')
 try:
   pipe = subprocess.Popen([ 'objdump', '-z', '-D', sys.argv[2] ],
-      bufsize=-1, stdout=subprocess.PIPE).stdout
+      bufsize=-1, stdout=subprocess.PIPE, universal_newlines=True).stdout
 except OSError as e:
   if e.errno == errno.ENOENT:
     print('''
