diff --git a/build.zig b/build.zig
index cb62d64da..87da0e695 100644
--- a/build.zig
+++ b/build.zig
@@ -666,6 +666,10 @@ fn addCmakeCfgOptionsToExe(
                     try addCxxKnownPath(b, cfg, exe, b.fmt("libstdc++.{s}", .{lib_suffix}), null, need_cpp_includes);
                 }
             },
+            .solaris => {
+                    try addCxxKnownPath(b, cfg, exe, b.fmt("libstdc++.{s}", .{lib_suffix}), null, need_cpp_includes);
+                    try addCxxKnownPath(b, cfg, exe, b.fmt("libgcc_eh.{s}", .{lib_suffix}), null, need_cpp_includes);
+            },
             else => {},
         }
     }
