diff -wpruN '--exclude=*.orig' a~/saslauthd/Makefile.am a/saslauthd/Makefile.am
--- a~/saslauthd/Makefile.am	1970-01-01 00:00:00
+++ a/saslauthd/Makefile.am	1970-01-01 00:00:00
@@ -41,7 +41,7 @@ DEFS            = @DEFS@ -DSASLAUTHD_CON
 dist-hook: saslauthd.8
 
 saslauthd.8: saslauthd.mdoc
-	nroff -mdoc $(srcdir)/saslauthd.mdoc > $(srcdir)/saslauthd.8
+	cp $(srcdir)/saslauthd.mdoc $(srcdir)/saslauthd.8
 
 install-data-local: saslauthd.8
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man8
diff -wpruN '--exclude=*.orig' a~/saslauthd/saslauthd.mdoc a/saslauthd/saslauthd.mdoc
--- a~/saslauthd/saslauthd.mdoc	1970-01-01 00:00:00
+++ a/saslauthd/saslauthd.mdoc	1970-01-01 00:00:00
@@ -38,37 +38,28 @@ authentication services to clients that
 SASL based authentication.
 .Pp
 .Nm
-should be
-started from the system boot scripts when going to
-multi-user mode. When running against a protected authentication
-database (e.g. the
-.Li shadow
-mechanism),
-it must be run as the superuser. Otherwise it is recommended to run
-daemon unprivileged as saslauth:saslauth, requiring the runtime directory
-to have root:saslauthd owner. You can do so by following
-these steps in machines using
-.Xr systemd 1
-:
-
-.Bl -enum -compact
-.It
-create directory
-.Pa /etc/systemd/system/saslauthd.service.d/
-.It
-create file
-.Pa /etc/systemd/system/saslauthd.service.d/user.conf
-with content
-.Bd -literal
-[Service]
-User=saslauth
-Group=saslauth
-
+is managed by the
+.Sy svc:/system/saslauthd
+service and runs in
+.Em pam
+mode by default.
+To change the authentication mechanism in use, configure the
+.Em config/mech
+property on the service.
+For example, to switch to the
+.Em shadow
+mechanism, issue the following commands:
+.Bd -literal -offset indent
+.Ic svccfg -s saslauthd setprop config/mech = shadow
+.Ic svcadm refresh saslauthd
+.Ic svcadm restart saslauthd
 .Ed
-.It
-Reload systemd service file: run
-.Dq systemctl daemon-reload
-.El
+.Pp
+To access the
+.Nm
+server socket, the caller must have root privileges or be a member of the
+.Em sasl
+group.
 .Ss Options
 Options named by lower\-case letters configure the server itself.
 Upper\-case options control the behavior of specific authentication
@@ -278,7 +269,7 @@ was never intended to be used in this ma
 .Bl -tag -width "/var/run/saslauthd/mux"
 .It Pa /var/run/saslauthd/mux
 The default communications socket.
-.It Pa /usr/local/etc/saslauthd.conf
+.It Pa /etc/opt/ooce/sasl/saslauthd.conf
 The default configuration file for ldap support.
 .El
 .Sh SEE ALSO
