diff -wpruN '--exclude=*.orig' a~/lib/auxprop.c a/lib/auxprop.c
--- a~/lib/auxprop.c	1970-01-01 00:00:00
+++ a/lib/auxprop.c	1970-01-01 00:00:00
@@ -822,8 +822,10 @@ int sasl_auxprop_add_plugin(const char *
     }
 
     if(result != SASL_OK) {
+        if(result != SASL_NOMECH) {
 	_sasl_log(NULL, SASL_LOG_ERR, "auxpropfunc error %s\n",
 		  sasl_errstring(result, NULL, NULL));
+        }
 	return result;
     }
 
diff -wpruN '--exclude=*.orig' a~/plugins/sql.c a/plugins/sql.c
--- a~/plugins/sql.c	1970-01-01 00:00:00
+++ a/plugins/sql.c	1970-01-01 00:00:00
@@ -122,7 +122,7 @@ static int _mysql_exec(void *conn, const
     result = mysql_store_result(conn);
     if (!result) {
 	/* umm nothing found */
-	utils->log(utils->conn, SASL_LOG_NOTE, "sql plugin: no result found");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql plugin: no result found");
 	return -1;
     }
 
@@ -131,7 +131,7 @@ static int _mysql_exec(void *conn, const
     if (!row_count) {
 	/* umm nothing found */
 	mysql_free_result(result);
-	utils->log(utils->conn, SASL_LOG_NOTE, "sql plugin: no result found");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql plugin: no result found");
 	return -1;
     }
     if (row_count > 1) {
@@ -144,7 +144,7 @@ static int _mysql_exec(void *conn, const
     row = mysql_fetch_row(result);
     if (!row || !row[0]) {
 	/* umm nothing found */
-	utils->log(utils->conn, SASL_LOG_NOTE, "sql plugin: no result found");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql plugin: no result found");
 	mysql_free_result(result);
 	return -1;
     }
@@ -298,7 +298,7 @@ static int _pgsql_exec(void *conn, const
     row_count = PQntuples(result);
     if (!row_count) {
 	/* umm nothing found */
-	utils->log(utils->conn, SASL_LOG_NOTE, "sql plugin: no result found");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql plugin: no result found");
 	PQclear(result);
 	return -1;
     }
@@ -426,7 +426,7 @@ static int _sqlite_exec(void *db, const
 
     if (result == NULL) {
 	/* umm nothing found */
-	utils->log(utils->conn, SASL_LOG_NOTE, "sql plugin: no result found");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql plugin: no result found");
 	return -1;
     }
 
@@ -569,7 +569,7 @@ static int _sqlite3_exec(void *db,
 
     if (result == NULL) {
 	/* umm nothing found */
-	utils->log(utils->conn, SASL_LOG_NOTE, "sql plugin: no result found");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql plugin: no result found");
 	return -1;
     }
 
@@ -1349,7 +1349,7 @@ int sql_auxprop_plug_init(const sasl_uti
     if (!settings->sql_engine->name) return SASL_NOMECH;
 
     if (!sql_exists(settings->sql_select)) {
-	utils->log(utils->conn, SASL_LOG_ERR, "sql_select option missing");
+	utils->log(utils->conn, SASL_LOG_DEBUG, "sql_select option missing");
 	utils->free(settings);	
 	return SASL_NOMECH;
     }
