diff -wpruN '--exclude=*.orig' a~/libmariadb/mariadb_config/CMakeLists.txt a/libmariadb/mariadb_config/CMakeLists.txt
--- a~/libmariadb/mariadb_config/CMakeLists.txt	2020-11-03 10:50:45.000000000 +0000
+++ a/libmariadb/mariadb_config/CMakeLists.txt	2020-11-09 16:30:55.395533712 +0000
@@ -59,7 +59,7 @@ ENDIF()
 # Installation
 #
 INSTALL(TARGETS mariadb_config
-        DESTINATION "bin"
+        DESTINATION "${INSTALL_BINDIR}"
         COMPONENT Development)
 
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libmariadb.pc
