diff --git a/listen.go b/listen.go
index 0cd3fabb..34812b54 100644
--- a/listen.go
+++ b/listen.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build !unix
+//go:build !unix || solaris
 
 package caddy
 
diff --git a/listen_unix.go b/listen_unix.go
index 34cd76c5..9ec65c39 100644
--- a/listen_unix.go
+++ b/listen_unix.go
@@ -15,7 +15,7 @@
 // Even though the filename ends in _unix.go, we still have to specify the
 // build constraint here, because the filename convention only works for
 // literal GOOS values, and "unix" is a shortcut unique to build tags.
-//go:build unix
+//go:build unix && !solaris
 
 package caddy
 
diff --git a/listen_unix_setopt.go b/listen_unix_setopt.go
index c9675f92..13ee7b83 100644
--- a/listen_unix_setopt.go
+++ b/listen_unix_setopt.go
@@ -1,4 +1,4 @@
-//go:build unix && !freebsd
+//go:build unix && !freebsd && !solaris
 
 package caddy
 
