diff -wpruN '--exclude=*.orig' a~/src/bootstrap/install.rs a/src/bootstrap/install.rs
--- a~/src/bootstrap/install.rs	1970-01-01 00:00:00
+++ a/src/bootstrap/install.rs	1970-01-01 00:00:00
@@ -106,7 +106,7 @@ fn install_sh(
         pkgname(builder, name)
     };
 
-    let mut cmd = Command::new("sh");
+    let mut cmd = Command::new("bash");
     cmd.current_dir(&empty_dir)
         .arg(sanitize_sh(&tmpdir(builder).join(&package_name).join("install.sh")))
         .arg(format!("--prefix={}", sanitize_sh(&prefix)))
