diff -wpruN '--exclude=*.orig' a~/conf.d/30-metric-aliases.conf a/conf.d/30-metric-aliases.conf
--- a~/conf.d/30-metric-aliases.conf	1970-01-01 00:00:00
+++ a/conf.d/30-metric-aliases.conf	1970-01-01 00:00:00
@@ -631,4 +631,13 @@ NOTE: The (URW)++ fonts mappings of gene
 	  </accept>
   </alias>
 
+        <!-- StarOffice formerly included Arial Narrow - Solaris includes Arial Narrow MT instead. -->
+        <alias>
+          <family>Arial Narrow</family>
+            <accept>
+              <family>Arial Narrow MT</family>
+              <family>Nimbus Sans L Condensed</family>
+            </accept>
+        </alias>
+
 </fontconfig>
