diff -pruN '--exclude=*.orig' ncurses-6.1-20180129~/configure ncurses-6.1-20180129/configure
--- ncurses-6.1-20180129~/configure	2018-01-20 00:27:18.000000000 +0000
+++ ncurses-6.1-20180129/configure	2018-02-01 18:23:53.482581860 +0000
@@ -25482,6 +25482,10 @@ TMPSED=\${TMPDIR:-/tmp}/headers.sed\$\$
 echo installing \$SRC in \$DST
 CF_EOF
 
+# Gratuitious newline because rarely, this script doesn't insert one here.
+# An extra newline won't hurt, however.
+echo "" >>headers.sh
+
 if test $WITH_CURSES_H = yes; then
 	cat >>headers.sh <<CF_EOF
 case \$DST in
