diff -wpruN --no-dereference '--exclude=*.orig' a~/dist/configure a/dist/configure
--- a~/dist/configure	1970-01-01 00:00:00
+++ a/dist/configure	1970-01-01 00:00:00
@@ -22485,6 +22485,7 @@ if test "$db_cv_atomic" = no; then
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <sys/atomic.h>
 int
 main ()
@@ -24192,6 +24193,9 @@ $as_echo_n "checking for 64-bit integral
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdio.h>
+#include <strings.h>
+
 int
 main ()
 {
