diff -wpruN --no-dereference '--exclude=*.orig' a~/build.zig a/build.zig
--- a~/build.zig	1970-01-01 00:00:00
+++ a/build.zig	1970-01-01 00:00:00
@@ -18,7 +18,7 @@ pub fn build(b: *std.Build) void {
     });
 
     exe.pie = pie;
-    exe.root_module.linkSystemLibrary("ncursesw", .{});
+    exe.root_module.linkSystemLibrary("ncurses", .{});
     exe.root_module.linkSystemLibrary("libzstd", .{});
     // https://github.com/ziglang/zig/blob/b52be973dfb7d1408218b8e75800a2da3dc69108/build.zig#L551-L554
     if (target.result.isDarwin()) {
@@ -43,7 +43,7 @@ pub fn build(b: *std.Build) void {
         .link_libc = true,
     });
     unit_tests.pie = pie;
-    unit_tests.root_module.linkSystemLibrary("ncursesw", .{});
+    unit_tests.root_module.linkSystemLibrary("ncurses", .{});
 
     const run_unit_tests = b.addRunArtifact(unit_tests);
 
diff -wpruN --no-dereference '--exclude=*.orig' a~/src/ui.zig a/src/ui.zig
--- a~/src/ui.zig	1970-01-01 00:00:00
+++ a/src/ui.zig	1970-01-01 00:00:00
@@ -11,7 +11,7 @@ pub const c = @cImport({
     @cDefine("_XOPEN_SOURCE", "1");
     @cInclude("stdio.h");
     @cInclude("string.h");
-    @cInclude("curses.h");
+    @cInclude("ncurses/curses.h");
     @cInclude("time.h");
     @cInclude("wchar.h");
     @cInclude("locale.h");
