diff -wpruN '--exclude=*.orig' a~/build.zig a/build.zig
--- a~/build.zig	1970-01-01 00:00:00
+++ a/build.zig	1970-01-01 00:00:00
@@ -49,7 +49,7 @@
 }
 
 pub fn linkNcurses(compile_step: *std.Build.CompileStep) void {
-    compile_step.linkSystemLibrary("ncursesw");
+    compile_step.linkSystemLibrary("ncurses");
     compile_step.linkLibC();
     compile_step.addCSourceFile(.{ .file = .{ .path = "src/ncurses_refs.c" }, .flags = &.{} });
 }
diff -wpruN '--exclude=*.orig' a~/src/ncurses_refs.c a/src/ncurses_refs.c
--- a~/src/ncurses_refs.c	1970-01-01 00:00:00
+++ a/src/ncurses_refs.c	1970-01-01 00:00:00
@@ -2,7 +2,7 @@
  * SPDX-License-Identifier: MIT
  */
 
-#include <curses.h>
+#include <ncurses/curses.h>
 
 /* Zig @cImport() has problems with the ACS_* macros. Two, in fact:
  *
diff -wpruN '--exclude=*.orig' a~/src/ui.zig a/src/ui.zig
--- a~/src/ui.zig	1970-01-01 00:00:00
+++ a/src/ui.zig	1970-01-01 00:00:00
@@ -11,7 +11,7 @@ pub const c = @cImport({
     @cDefine("_XOPEN_SOURCE", "1");
     @cInclude("stdio.h");
     @cInclude("string.h");
-    @cInclude("curses.h");
+    @cInclude("ncurses/curses.h");
     @cInclude("time.h");
     @cInclude("wchar.h");
     @cInclude("locale.h");
