diff -pruN '--exclude=*.orig' fio-fio-3.2~/tools/plot/fio2gnuplot fio-fio-3.2/tools/plot/fio2gnuplot
--- fio-fio-3.2~/tools/plot/fio2gnuplot	2017-11-03 22:23:49.000000000 +0000
+++ fio-fio-3.2/tools/plot/fio2gnuplot	2017-11-07 17:01:12.124463527 +0000
@@ -401,9 +401,9 @@ def main(argv):
     force_keep_temp_files=False
 
     if not os.path.isfile(gpm_dir+'math.gpm'):
-	    gpm_dir="/usr/local/share/fio/"
+	    gpm_dir="/usr/share/fio/"
     	    if not os.path.isfile(gpm_dir+'math.gpm'):
-		    print "Looks like fio didn't get installed properly as no gpm files found in '/usr/share/fio' or '/usr/local/share/fio'\n"
+		    print "Looks like fio didn't get installed properly as no gpm files found in '/usr/share/fio'\n"
 		    sys.exit(3)
 
     try:
