diff -pruN '--exclude=*.orig' fio-fio-3.1~/Makefile fio-fio-3.1/Makefile
--- fio-fio-3.1~/Makefile	2017-09-28 12:23:20.000000000 +0000
+++ fio-fio-3.1/Makefile	2017-09-28 14:05:26.435439450 +0000
@@ -295,16 +295,19 @@ endif
 endif
 
 ifeq ($(CONFIG_TARGET_OS), SunOS)
-	INSTALL = ginstall
+	INSTALL = /usr/gnu/bin/install
 else
 	INSTALL = install
 endif
-prefix = $(INSTALL_PREFIX)
+prefix = /usr
 bindir = $(prefix)/bin
 
 ifeq ($(CONFIG_TARGET_OS), Darwin)
 mandir = /usr/share/man
 sharedir = /usr/share/fio
+else ifeq ($(CONFIG_TARGET_OS), SunOS)
+mandir = $(prefix)/share/man
+sharedir = $(prefix)/share/fio
 else
 mandir = $(prefix)/man
 sharedir = $(prefix)/share/fio
