diff -pruN '--exclude=*.orig' tmux-2.6~/tmux.h tmux-2.6/tmux.h
--- tmux-2.6~/tmux.h	2017-08-30 20:23:42.000000000 +0000
+++ tmux-2.6/tmux.h	2017-10-05 14:06:39.104418928 +0000
@@ -25,6 +25,12 @@
 #include <sys/uio.h>
 
 #include <event.h>
+#include <event_compat.h>
+#include <event_struct.h>
+#include <buffer.h>
+#include <bufferevent.h>
+#include <bufferevent_struct.h>
+#include <buffer_compat.h>
 #include <limits.h>
 #include <stdarg.h>
 #include <stdio.h>
