diff -wpruN --no-dereference '--exclude=*.orig' a~/lib/Driver/ToolChain.cpp a/lib/Driver/ToolChain.cpp
--- a~/lib/Driver/ToolChain.cpp	1970-01-01 00:00:00
+++ a/lib/Driver/ToolChain.cpp	1970-01-01 00:00:00
@@ -1233,9 +1233,7 @@ void ToolChain::AddClangCXXStdlibIsystem
 }
 
 bool ToolChain::ShouldLinkCXXStdlib(const llvm::opt::ArgList &Args) const {
-  return getDriver().CCCIsCXX() &&
-         !Args.hasArg(options::OPT_nostdlib, options::OPT_nodefaultlibs,
-                      options::OPT_nostdlibxx);
+  return getDriver().CCCIsCXX() && !Args.hasArg(options::OPT_nostdlibxx);
 }
 
 void ToolChain::AddCXXStdlibLibArgs(const ArgList &Args,
diff -wpruN --no-dereference '--exclude=*.orig' a~/lib/Driver/ToolChains/Solaris.cpp a/lib/Driver/ToolChains/Solaris.cpp
--- a~/lib/Driver/ToolChains/Solaris.cpp	1970-01-01 00:00:00
+++ a/lib/Driver/ToolChains/Solaris.cpp	1970-01-01 00:00:00
@@ -207,6 +207,9 @@ void solaris::Linker::ConstructJob(Compi
   bool NeedsSanitizerDeps = addSanitizerRuntimes(ToolChain, Args, CmdArgs);
   AddLinkerInputs(ToolChain, Inputs, Args, CmdArgs, JA);
 
+  if (ToolChain.ShouldLinkCXXStdlib(Args) && !Args.hasArg(options::OPT_r))
+    ToolChain.AddCXXStdlibLibArgs(Args, CmdArgs);
+
   if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nodefaultlibs,
                    options::OPT_r)) {
     // Use the static OpenMP runtime with -static-openmp
@@ -215,8 +218,6 @@ void solaris::Linker::ConstructJob(Compi
     addOpenMPRuntime(CmdArgs, ToolChain, Args, StaticOpenMP);
 
     if (D.CCCIsCXX()) {
-      if (ToolChain.ShouldLinkCXXStdlib(Args))
-        ToolChain.AddCXXStdlibLibArgs(Args, CmdArgs);
       CmdArgs.push_back("-lm");
     }
     // Silence warnings when linking C code with a C++ '-stdlib' argument.
@@ -419,3 +420,16 @@ void Solaris::addLibStdCxxIncludePaths(
                            TripleStr, Multilib.includeSuffix(), DriverArgs,
                            CC1Args);
 }
+
+void Solaris::AddCXXStdlibLibArgs(const ArgList &Args,
+                                  ArgStringList &CmdArgs) const {
+  Generic_ELF::AddCXXStdlibLibArgs(Args, CmdArgs);
+
+  if (GetCXXStdlibType(Args) != ToolChain::CST_Libstdcxx)
+    return;
+
+  StringRef LibSuffix = getSolarisLibSuffix(getTriple());
+  std::string MajVer = GCCInstallation.getVersion().MajorStr;
+  CmdArgs.push_back(Args.MakeArgString(StringRef("-R")
+    + "/usr/gcc/" + MajVer + "/lib" + LibSuffix));
+}
diff -wpruN --no-dereference '--exclude=*.orig' a~/lib/Driver/ToolChains/Solaris.h a/lib/Driver/ToolChains/Solaris.h
--- a~/lib/Driver/ToolChains/Solaris.h	1970-01-01 00:00:00
+++ a/lib/Driver/ToolChains/Solaris.h	1970-01-01 00:00:00
@@ -64,6 +64,9 @@ public:
   addLibStdCxxIncludePaths(const llvm::opt::ArgList &DriverArgs,
                            llvm::opt::ArgStringList &CC1Args) const override;
 
+  void AddCXXStdlibLibArgs(const llvm::opt::ArgList &Args,
+                           llvm::opt::ArgStringList &CmdArgs) const override;
+
   SanitizerMask getSupportedSanitizers() const override;
 
   const char *getDefaultLinker() const override;
