diff -wpruN --no-dereference '--exclude=*.orig' a~/src/syscfg/lock-obj-pub.aarch64-unknown-solaris2.11.h a/src/syscfg/lock-obj-pub.aarch64-unknown-solaris2.11.h
--- a~/src/syscfg/lock-obj-pub.aarch64-unknown-solaris2.11.h	1970-01-01 00:00:00
+++ a/src/syscfg/lock-obj-pub.aarch64-unknown-solaris2.11.h	1970-01-01 00:00:00
@@ -0,0 +1,23 @@
+## lock-obj-pub.aarch64-unknown-solaris2.11.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,88,77, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
