diff -wpruN '--exclude=*.orig' a~/gsoap/stdsoap2.c a/gsoap/stdsoap2.c
--- a~/gsoap/stdsoap2.c	1970-01-01 00:00:00
+++ a/gsoap/stdsoap2.c	1970-01-01 00:00:00
@@ -3931,7 +3931,7 @@ soap_rand()
   int r;
   if (!soap_ssl_init_done)
     soap_ssl_init();
-  RAND_pseudo_bytes((unsigned char*)&r, sizeof(int));
+  RAND_bytes((unsigned char*)&r, sizeof(int));
   return r;
 }
 #endif
@@ -4394,7 +4394,11 @@ ssl_auth_init(struct soap *soap)
     if (SSL_CTX_need_tmp_RSA(soap->ctx))
 #endif
     {
-      RSA *rsa = RSA_generate_key(SOAP_SSL_RSA_BITS, RSA_F4, NULL, NULL);
+      RSA *rsa = RSA_new();
+      BIGNUM *e = BN_new();
+      BN_set_word(e, RSA_F4);
+      RSA_generate_key_ex(rsa, SOAP_SSL_RSA_BITS, e, NULL);
+      BN_free(e);
       if (!rsa || !SSL_CTX_set_tmp_rsa(soap->ctx, rsa))
       {
         if (rsa)
@@ -4414,7 +4418,10 @@ ssl_auth_init(struct soap *soap)
 #if defined(VXWORKS)
       DH_generate_parameters_ex(dh, n, 2/*or 5*/, NULL); /* vxWorks compatible */
 #else
-      dh = DH_generate_parameters(n, 2/*or 5*/, NULL, NULL);
+{
+      dh = DH_new();
+      DH_generate_parameters_ex(dh, n, 2/*or 5*/, NULL);
+}
 #endif
     else
     {
diff -wpruN '--exclude=*.orig' a~/gsoap/stdsoap2.cpp a/gsoap/stdsoap2.cpp
--- a~/gsoap/stdsoap2.cpp	1970-01-01 00:00:00
+++ a/gsoap/stdsoap2.cpp	1970-01-01 00:00:00
@@ -3931,7 +3931,7 @@ soap_rand()
   int r;
   if (!soap_ssl_init_done)
     soap_ssl_init();
-  RAND_pseudo_bytes((unsigned char*)&r, sizeof(int));
+  RAND_bytes((unsigned char*)&r, sizeof(int));
   return r;
 }
 #endif
@@ -4394,7 +4394,11 @@ ssl_auth_init(struct soap *soap)
     if (SSL_CTX_need_tmp_RSA(soap->ctx))
 #endif
     {
-      RSA *rsa = RSA_generate_key(SOAP_SSL_RSA_BITS, RSA_F4, NULL, NULL);
+      RSA *rsa = RSA_new();
+      BIGNUM *e = BN_new();
+      BN_set_word(e, RSA_F4);
+      RSA_generate_key_ex(rsa, SOAP_SSL_RSA_BITS, e, NULL);
+      BN_free(e);
       if (!rsa || !SSL_CTX_set_tmp_rsa(soap->ctx, rsa))
       {
         if (rsa)
@@ -4414,7 +4418,10 @@ ssl_auth_init(struct soap *soap)
 #if defined(VXWORKS)
       DH_generate_parameters_ex(dh, n, 2/*or 5*/, NULL); /* vxWorks compatible */
 #else
-      dh = DH_generate_parameters(n, 2/*or 5*/, NULL, NULL);
+{
+      dh = DH_new();
+      DH_generate_parameters_ex(dh, n, 2/*or 5*/, NULL);
+}
 #endif
     else
     {
@@ -5987,7 +5994,7 @@ again:
             name = X509_NAME_ENTRY_get_data(X509_NAME_get_entry(subj, i));
             if (name)
             {
-              const char *tmp = (const char*)ASN1_STRING_data(name);
+              const char *tmp = (const char*)ASN1_STRING_get0_data(name);
               if (!soap_tag_cmp(host, tmp))
               {
                 ok = 1;
diff -wpruN '--exclude=*.orig' a~/gsoap/stdsoap2.h a/gsoap/stdsoap2.h
--- a~/gsoap/stdsoap2.h	1970-01-01 00:00:00
+++ a/gsoap/stdsoap2.h	1970-01-01 00:00:00
@@ -934,6 +934,7 @@ extern intmax_t __strtoull(const char*,
 # include <openssl/rand.h>
 # include <openssl/ssl.h>
 # include <openssl/x509v3.h>
+# include <openssl/rsa.h>
 # ifndef ALLOW_OLD_VERSIONS
 #  if (OPENSSL_VERSION_NUMBER < 0x00905100L)
 #   error "Must use OpenSSL 0.9.6 or later"
