diff --git a/m4/ldap.m4 b/m4/ldap.m4
index df7b71c..20a5f5d 100644
--- a/m4/ldap.m4
+++ b/m4/ldap.m4
@@ -99,7 +99,7 @@ AS_HELP_STRING([--with-ldap@<:@=DIR@:>@],[Include LDAP support @<:@default=no@:>
         else
                 if test -f $_libldap_with/include/ldap.h; then
                         LDAP_INCDIR=$_libldap_with/include
-                        LDAP_LIBDIR=$_libldap_with/lib
+                        LDAP_LIBDIR=$_libldap_with/lib/amd64
                         _ldap_dir_lib="$LDAP_LIBDIR"
 
                         found_ldap="yes"
@@ -120,7 +120,7 @@ AS_HELP_STRING([--with-ldap@<:@=DIR@:>@],[Include LDAP support @<:@default=no@:>
                         AC_DEFINE(HAVE_LDAP_SOURCEIP, 1, [Define to 1 if SourceIP is supported.])
                 fi
 
-                LDAP_LIBS="-lldap -llber $LDAP_LIBS"
+                LDAP_LIBS="-lldap_r -llber $LDAP_LIBS"
 
                 if test "x$enable_static" = "xyes"; then
                         LDAP_LIBS="$LDAP_LIBS -lgnutls -lpthread -lsasl2"
@@ -129,7 +129,7 @@ AS_HELP_STRING([--with-ldap@<:@=DIR@:>@],[Include LDAP support @<:@default=no@:>
                         test "x$static_linking_support" = "xno" -a -z "$_ldap_dir_lib" && AC_MSG_ERROR(["Compiler not support statically linked libs from default folders"])
 
                         if test "x$static_linking_support" = "xno"; then
-                                LDAP_LIBS=`echo "$LDAP_LIBS"|sed "s|-lldap|$_ldap_dir_lib/libldap.a|g"|sed "s|-llber|$_ldap_dir_lib/liblber.a|g"`
+                                LDAP_LIBS=`echo "$LDAP_LIBS"|sed "s|-lldap_r|$_ldap_dir_lib/libldap.a|g"|sed "s|-llber|$_ldap_dir_lib/liblber.a|g"`
                         fi
 
                         # without SSL and SASL
diff --git a/m4/libcurl.m4 b/m4/libcurl.m4
index 5ff8abe..41bce3f 100644
--- a/m4/libcurl.m4
+++ b/m4/libcurl.m4
@@ -157,7 +157,7 @@ AS_HELP_STRING([--with-libcurl@<:@=FILE@:>@],[use cURL package @<:@default=no@:>
 												-lssl|-lcrypto)
 													test "x$want_openssl" = "xyes" && i="$OPENSSL_LIBS"
 											;;
-												-lldap|-lldap_r|-llber)
+												-lldap_r|-llber)
 													test "x$want_ldap" = "xyes" && i="$LDAP_LIBS"
 											;;
 												-l*)
