diff --git a/conf/zabbix_server.conf b/conf/zabbix_server.conf
index 0e19248..f5bee74 100644
--- a/conf/zabbix_server.conf
+++ b/conf/zabbix_server.conf
@@ -35,7 +35,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_server.log
+LogFile=/var/opt/ooce/zabbix/zabbix_server.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -44,7 +44,7 @@ LogFile=/tmp/zabbix_server.log
 # Mandatory: no
 # Range: 0-1024
 # Default:
-# LogFileSize=1
+LogFileSize=10
 
 ### Option: DebugLevel
 #	Specifies debug level:
@@ -65,7 +65,7 @@ LogFile=/tmp/zabbix_server.log
 #
 # Mandatory: no
 # Default:
-# PidFile=/tmp/zabbix_server.pid
+PidFile=/var/opt/ooce/zabbix/zabbix_server.pid
 
 ### Option: SocketDir
 #	IPC socket directory.
@@ -74,6 +74,7 @@ LogFile=/tmp/zabbix_server.log
 # Mandatory: no
 # Default:
 # SocketDir=/tmp
+SocketDir=/var/opt/ooce/zabbix
 
 ### Option: DBHost
 #	Database host name.
@@ -417,6 +418,7 @@ DBUser=zabbix
 # Mandatory: no
 # Default:
 # SNMPTrapperFile=/tmp/zabbix_traps.tmp
+SNMPTrapperFile=/var/opt/ooce/zabbix/zabbix_traps.tmp
 
 ### Option: StartSNMPTrapper
 #	If 1, SNMP trapper process is started.
@@ -601,7 +603,7 @@ Timeout=4
 #
 # Mandatory: no
 # Default:
-# FpingLocation=/usr/sbin/fping
+FpingLocation=/opt/ooce/sbin/fping
 
 ### Option: Fping6Location
 #	Location of fping6.
@@ -1153,7 +1155,7 @@ EnableGlobalScripts=0
 
 ### Option: Include
 #	You may include individual files or all files in a directory in the configuration file.
-#	Installing Zabbix will create include directory in /usr/local/etc, unless modified during the compile time.
+#	Installing Zabbix will create include directory in /etc/opt/ooce/zabbix, unless modified during the compile time.
 #
 # Mandatory: no
 # Default:
@@ -1162,3 +1164,6 @@ EnableGlobalScripts=0
 # Include=/usr/local/etc/zabbix_server.general.conf
 # Include=/usr/local/etc/zabbix_server.conf.d/
 # Include=/usr/local/etc/zabbix_server.conf.d/*.conf
+# Include=/etc/opt/ooce/zabbix/zabbix_server.general.conf
+# Include=/etc/opt/ooce/zabbix/zabbix_server.conf.d/
+Include=/etc/opt/ooce/zabbix/zabbix_server.conf.d/*.conf
