diff -pruN '--exclude=*.orig' expat-2.2.5~/configure expat-2.2.5/configure
--- expat-2.2.5~/configure	2017-10-31 15:58:33.000000000 +0000
+++ expat-2.2.5/configure	2017-11-07 13:09:25.505605793 +0000
@@ -16381,9 +16381,9 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 if test "$GCC" = yes ; then
                         OLDCFLAGS="$CFLAGS -Wall -Wmissing-prototypes -Wstrict-prototypes"
-    CFLAGS="$OLDCFLAGS -fexceptions"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -fexceptions" >&5
-$as_echo_n "checking whether $CC accepts -fexceptions... " >&6; }
+    CFLAGS="$OLDCFLAGS -funwind-tables"
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -funwind-tables" >&5
+$as_echo_n "checking whether $CC accepts -funwind-tables... " >&6; }
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
diff -pruN '--exclude=*.orig' expat-2.2.5~/configure.ac expat-2.2.5/configure.ac
--- expat-2.2.5~/configure.ac	2017-10-27 20:04:59.000000000 +0000
+++ expat-2.2.5/configure.ac	2017-11-07 13:09:25.505787046 +0000
@@ -69,13 +69,13 @@ AC_PROG_INSTALL
 
 if test "$GCC" = yes ; then
     dnl
-    dnl Be careful about adding the -fexceptions option; some versions of
+    dnl Be careful about adding the -funwind-tables option; some versions of
     dnl GCC don't support it and it causes extra warnings that are only
     dnl distracting; avoid.
     dnl
     OLDCFLAGS="$CFLAGS -Wall -Wmissing-prototypes -Wstrict-prototypes"
-    CFLAGS="$OLDCFLAGS -fexceptions"
-    AC_MSG_CHECKING(whether $CC accepts -fexceptions)
+    CFLAGS="$OLDCFLAGS -funwind-tables"
+    AC_MSG_CHECKING(whether $CC accepts -funwind-tables)
     AC_TRY_LINK( , ,
                    AC_MSG_RESULT(yes),
                    AC_MSG_RESULT(no); CFLAGS="$OLDCFLAGS")
