diff -pruN '--exclude=*.orig' patch-2.7.6~/tests/git-cleanup patch-2.7.6/tests/git-cleanup
--- patch-2.7.6~/tests/git-cleanup	2018-02-03 13:41:49.000000000 +0000
+++ patch-2.7.6/tests/git-cleanup	2018-02-07 23:53:22.940660075 +0000
@@ -36,8 +36,8 @@ BAD PATCH
 EOF
 
 echo 1 > f
-echo -n '' > g
-echo -n '' > h
+printf '' > g
+printf '' > h
 
 check 'patch -f -i 1.diff || echo status: $?' <<EOF
 patching file f
