diff -pruN '--exclude=*.orig' binutils-2.29.1~/gold/x86_64.cc binutils-2.29.1/gold/x86_64.cc
--- binutils-2.29.1~/gold/x86_64.cc	2017-07-10 09:54:42.000000000 +0000
+++ binutils-2.29.1/gold/x86_64.cc	2018-01-10 21:24:04.500903728 +0000
@@ -1259,7 +1259,11 @@ const Target::Target_info Target_x86_64<
   true,			// is_default_stack_executable
   true,			// can_icf_inline_merge_sections
   '\0',			// wrap_char
+#if defined(__sun__)
+  "/lib/64/ld.so.1",
+#else
   "/lib/ld64.so.1",     // program interpreter
+#endif
   0x400000,		// default_text_segment_address
   0x1000,		// abi_pagesize (overridable by -z max-page-size)
   0x1000,		// common_pagesize (overridable by -z common-page-size)
