diff -wpruN '--exclude=*.orig' a~/libr/anal/p/anal_arm_cs.c a/libr/anal/p/anal_arm_cs.c
--- a~/libr/anal/p/anal_arm_cs.c	1970-01-01 00:00:00
+++ a/libr/anal/p/anal_arm_cs.c	1970-01-01 00:00:00
@@ -1361,13 +1361,13 @@ static int analop64_esil(RAnal *a, RAnal
 	case ARM64_INS_AND:
 		OPCALL("&");
 		break;
-	case ARM64_INS_NAND:
-		OPCALL_NEG("&");
-		break;
 	case ARM64_INS_ORR:
 		OPCALL("|");
 		break;
 #if CS_API_MAJOR > 4	
+	case ARM64_INS_NAND:
+		OPCALL_NEG("&");
+		break;
 	case ARM64_INS_ADDS:
 	case ARM64_INS_ADCS:
 		OPCALL("+");
