From 3b0679ed6f7bdf0d2b2159395887b7ef19d39f03 Mon Sep 17 00:00:00 2001
From: "Joshua M. Clulow" <josh@sysmgr.org>
Date: Mon, 27 Dec 2021 16:08:21 -0800
Subject: [PATCH] sunos: errno mishandled in sunos_submit_transfer()

---
 libusb/os/sunos_usb.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libusb/os/sunos_usb.c b/libusb/os/sunos_usb.c
index 28b167ff..7c2f615e 100644
--- a/libusb/os/sunos_usb.c
+++ b/libusb/os/sunos_usb.c
@@ -1379,8 +1379,7 @@ sunos_submit_transfer(struct usbi_transfer *itransfer)
 
 	err = sunos_check_device_and_status_open(hdl,
 	    transfer->endpoint, transfer->type);
-	if (err < 0) {
-
+	if (err != 0) {
 		return (_errno_to_libusb(err));
 	}
 
