diff -pruN '--exclude=*.orig' coreutils-8.28~/src/install.c coreutils-8.28/src/install.c
--- coreutils-8.28~/src/install.c	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/src/install.c	2017-10-21 11:40:15.887970444 +0000
@@ -635,8 +635,8 @@ Usage: %s [OPTION]... [-T] SOURCE DEST\n
 \n\
 This install program copies files (often just compiled) into destination\n\
 locations you choose.  If you want to download and install a ready-to-use\n\
-package on a GNU/Linux system, you should instead be using a package manager\n\
-like yum(1) or apt-get(1).\n\
+package on an OmniOS system, you should instead be using a package manager\n\
+like pkg(1).\n\
 \n\
 In the first three forms, copy SOURCE to DEST or multiple SOURCE(s) to\n\
 the existing DIRECTORY, while setting permission modes and owner/group.\n\
