diff -pruN '--exclude=*.orig' coreutils-8.28~/Makefile.in coreutils-8.28/Makefile.in
--- coreutils-8.28~/Makefile.in	2017-09-02 02:25:09.000000000 +0000
+++ coreutils-8.28/Makefile.in	2017-10-21 11:40:17.540512941 +0000
@@ -2644,7 +2644,7 @@ distuninstallcheck_listfiles = find . -t
 am__distuninstallcheck_listfiles = $(distuninstallcheck_listfiles) \
   | sed 's|^\./|$(prefix)/|' | grep -v '$(infodir)/dir$$'
 distcleancheck_listfiles = find . -type f -print
-pkglibexecdir = @pkglibexecdir@
+pkglibexecdir = @libexecdir@
 
 # Use 'ginstall' in the definition of PROGRAMS and in dependencies to avoid
 # confusion with the 'install' target.  The install rule transforms 'ginstall'
diff -pruN '--exclude=*.orig' coreutils-8.28~/gnulib-tests/Makefile.in coreutils-8.28/gnulib-tests/Makefile.in
--- coreutils-8.28~/gnulib-tests/Makefile.in	2017-09-01 07:17:11.000000000 +0000
+++ coreutils-8.28/gnulib-tests/Makefile.in	2017-10-21 11:40:17.542048931 +0000
@@ -2404,7 +2404,7 @@ am__relativize = \
     dir1=`echo "$$dir1" | sed -e "$$sed_rest"`; \
   done; \
   reldir="$$dir2"
-pkglibexecdir = @pkglibexecdir@
+pkglibexecdir = @libexecdir@
 ACLOCAL = @ACLOCAL@
 ALLOCA = @ALLOCA@
 ALLOCA_H = @ALLOCA_H@
