diff -pruN '--exclude=*.orig' coreutils-8.28~/man/mkfifo.x coreutils-8.28/man/mkfifo.x
--- coreutils-8.28~/man/mkfifo.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/mkfifo.x	2017-10-21 11:40:16.373199566 +0000
@@ -3,4 +3,4 @@ mkfifo \- make FIFOs (named pipes)
 [DESCRIPTION]
 .\" Add any additional description here
 [SEE ALSO]
-mkfifo(3)
+mkfifo(3C)
diff -pruN '--exclude=*.orig' coreutils-8.28~/man/mktemp.x coreutils-8.28/man/mktemp.x
--- coreutils-8.28~/man/mktemp.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/mktemp.x	2017-10-21 11:40:16.373579431 +0000
@@ -3,4 +3,4 @@ mktemp \- create a temporary file or dir
 [DESCRIPTION]
 .\" Add any additional description here
 [SEE ALSO]
-mkstemp(3), mkdtemp(3), mktemp(3)
+mkstemp(3C), mkdtemp(3C), mktemp(3C)
diff -pruN '--exclude=*.orig' coreutils-8.28~/man/printf.x coreutils-8.28/man/printf.x
--- coreutils-8.28~/man/printf.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/printf.x	2017-10-21 11:40:16.373855076 +0000
@@ -3,4 +3,4 @@ printf \- format and print data
 [DESCRIPTION]
 .\" Add any additional description here
 [SEE ALSO]
-printf(3)
+printf(3C)
diff -pruN '--exclude=*.orig' coreutils-8.28~/man/pwd.x coreutils-8.28/man/pwd.x
--- coreutils-8.28~/man/pwd.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/pwd.x	2017-10-21 11:40:16.374133469 +0000
@@ -3,4 +3,4 @@ pwd \- print name of current/working dir
 [DESCRIPTION]
 .\" Add any additional description here
 [SEE ALSO]
-getcwd(3)
+getcwd(3C)
diff -pruN '--exclude=*.orig' coreutils-8.28~/man/rm.x coreutils-8.28/man/rm.x
--- coreutils-8.28~/man/rm.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/rm.x	2017-10-21 11:40:16.374438082 +0000
@@ -28,4 +28,4 @@ prompts the user for whether to remove t
 not affirmative, the file is skipped.
 .SH OPTIONS
 [SEE ALSO]
-unlink(1), unlink(2), chattr(1), shred(1)
+unlink(1), unlink(2), shred(1)
diff -pruN '--exclude=*.orig' coreutils-8.28~/man/sleep.x coreutils-8.28/man/sleep.x
--- coreutils-8.28~/man/sleep.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/sleep.x	2017-10-21 11:40:16.374750727 +0000
@@ -3,4 +3,4 @@ sleep \- delay for a specified amount of
 [DESCRIPTION]
 .\" Add any additional description here
 [SEE ALSO]
-sleep(3)
+sleep(3C)
diff -pruN '--exclude=*.orig' coreutils-8.28~/man/truncate.x coreutils-8.28/man/truncate.x
--- coreutils-8.28~/man/truncate.x	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/man/truncate.x	2017-10-21 11:40:16.429724260 +0000
@@ -3,4 +3,4 @@ truncate \- shrink or extend the size of
 [DESCRIPTION]
 .\" Add any additional description here
 [SEE ALSO]
-dd(1), truncate(2), ftruncate(2)
+dd(1), truncate(3C), ftruncate(3C)
