diff -pruN '--exclude=*.orig' coreutils-8.28~/src/who.c coreutils-8.28/src/who.c
--- coreutils-8.28~/src/who.c	2017-09-01 07:11:03.000000000 +0000
+++ coreutils-8.28/src/who.c	2017-10-21 11:40:17.230504468 +0000
@@ -517,8 +517,8 @@ print_runlevel (const STRUCT_UTMP *utmp_
   unsigned char curr = UT_PID (utmp_ent) % 256;
 
   if (!runlevline)
-    runlevline = xmalloc (strlen (_("run-level")) + 3);
-  sprintf (runlevline, "%s %c", _("run-level"), curr);
+    runlevline = xmalloc (sizeof (utmp_ent->ut_line));
+  sprintf (runlevline, "%s", utmp_ent->ut_line);
 
   if (!comment)
     comment = xmalloc (strlen (_("last=")) + 2);
