From 1828bbcc2fcc9d0c6d809c49db0d9fb5eed2444c Mon Sep 17 00:00:00 2001
From: Dominik Hassler <hadfl@omnios.org>
Date: Fri, 5 Apr 2024 18:19:37 +0000
Subject: [PATCH] initial illumos support

---
 Cargo.lock | 3 +--
 Cargo.toml | 2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 86227c9772..74ac51e615 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1837,8 +1837,7 @@ dependencies = [
 [[package]]
 name = "webbrowser"
 version = "0.8.13"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "d1b04c569c83a9bb971dd47ec6fd48753315f4bf989b9b04a2e7ca4d7f0dc950"
+source = "git+https://github.com/hadfl/webbrowser-rs.git?branch=illumos_v0.8.13#ffaf6747c41ac944e1dfdb3c592807198b11e085"
 dependencies = [
  "core-foundation",
  "home",
diff --git a/Cargo.toml b/Cargo.toml
index b0622f40ae..975c534f59 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -83,7 +83,7 @@ toml = "0.8.12"
 unindent = "0.2.3"
 walkdir = "2.5.0"
 wasmparser = "0.201.0"
-webbrowser = "0.8.13"
+webbrowser = { git = "https://github.com/hadfl/webbrowser-rs.git", branch="illumos_v0.8.13" }
 
 tree-sitter = { version = "0.22.2", path = "./lib" }
 tree-sitter-loader = { version = "0.22.2", path = "./cli/loader" }
