diff -wpruN --no-dereference '--exclude=*.orig' a~/tools/libclang/CMakeLists.txt a/tools/libclang/CMakeLists.txt
--- a~/tools/libclang/CMakeLists.txt	1970-01-01 00:00:00
+++ a/tools/libclang/CMakeLists.txt	1970-01-01 00:00:00
@@ -97,7 +97,7 @@ if(MSVC)
   set(LLVM_EXPORTED_SYMBOL_FILE)
 endif()
 
-if (UNIX AND NOT APPLE)
+if (UNIX AND NOT APPLE AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
   set(LLVM_EXPORTED_SYMBOL_FILE)
   set(USE_VERSION_SCRIPT ${LLVM_HAVE_LINK_VERSION_SCRIPT})
 endif()
