diff -ru openssh-7.1p1.orig/sshd_config openssh-7.1p1/sshd_config
diff -pruN '--exclude=*.orig' openssh-7.5p1~/sshd_config openssh-7.5p1/sshd_config
--- openssh-7.5p1~/sshd_config	2017-03-20 03:39:27.000000000 +0000
+++ openssh-7.5p1/sshd_config	2017-10-06 12:34:44.215245761 +0000
@@ -30,7 +30,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin prohibit-password
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
@@ -89,7 +89,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #UseLogin no
