diff -wpruN '--exclude=*.orig' a~/programs/util.c a/programs/util.c
--- a~/programs/util.c	1970-01-01 00:00:00
+++ a/programs/util.c	1970-01-01 00:00:00
@@ -905,7 +905,7 @@
     return numPhysicalCores;
 }
 
-#elif defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__) || defined(__CYGWIN__)
+#elif defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__) || defined(__CYGWIN__) || defined(__sun__)
 
 /* Use POSIX sysconf
  * see: man 3 sysconf */
