diff -pruN '--exclude=*.orig' perl-5.26.1~/t/op/getppid.t perl-5.26.1/t/op/getppid.t
--- perl-5.26.1~/t/op/getppid.t	2017-07-19 00:50:18.000000000 +0000
+++ perl-5.26.1/t/op/getppid.t	2017-11-08 19:35:50.816444845 +0000
@@ -11,6 +11,10 @@ BEGIN {
     chdir 't' if -d 't';
     require './test.pl';
     set_up_inc( qw(../lib) );
+    if ($^O eq 'solaris' && `zonename` ne 'global') {
+        print "1..0 # Skip: not in illumos global zone\n";
+        exit (0);
+    }
 }
 
 use strict;
