diff -wpruN '--exclude=*.orig' a~/vagrant-installers/substrate/launcher/main.go a/vagrant-installers/substrate/launcher/main.go
--- a~/vagrant-installers/substrate/launcher/main.go	1970-01-01 00:00:00
+++ a/vagrant-installers/substrate/launcher/main.go	1970-01-01 00:00:00
@@ -14,7 +14,7 @@ import (
 	"strings"
 	"syscall"
 
-	"github.com/mitchellh/osext"
+	"github.com/kardianos/osext"
 )
 
 const envPrefix = "VAGRANT_OLD_ENV"
