diff -wpruN --no-dereference '--exclude=*.orig' a~/src/gl/tests/localename-unsafe.c a/src/gl/tests/localename-unsafe.c
--- a~/src/gl/tests/localename-unsafe.c	1970-01-01 00:00:00
+++ a/src/gl/tests/localename-unsafe.c	1970-01-01 00:00:00
@@ -48,7 +48,7 @@
 # if defined __sun
 #  if HAVE_GETLOCALENAME_L
 /* Solaris >= 12.  */
-extern char * getlocalename_l(int, locale_t);
+extern const char * getlocalename_l(int, locale_t);
 #  elif HAVE_SOLARIS114_LOCALES
 #   include <sys/localedef.h>
 #  endif
