diff -pruN '--exclude=*.orig' screen-4.6.1~/Makefile.in screen-4.6.1/Makefile.in
--- screen-4.6.1~/Makefile.in	2017-07-10 19:26:25.000000000 +0000
+++ screen-4.6.1/Makefile.in	2017-10-18 10:04:34.021872031 +0000
@@ -17,7 +17,7 @@ datarootdir = @datarootdir@
 datadir = @datadir@
 
 # don't forget to change mandir and infodir in doc/Makefile.
-bindir  = $(exec_prefix)/bin
+bindir  = @bindir@
 
 VERSION = @VERSION@
 SCREEN = screen-$(VERSION)
