diff -pruN '--exclude=*.orig' Python-2.7.14~/Modules/python.c Python-2.7.14/Modules/python.c
--- Python-2.7.14~/Modules/python.c	2017-09-16 17:38:35.000000000 +0000
+++ Python-2.7.14/Modules/python.c	2017-09-23 22:33:28.270473428 +0000
@@ -6,6 +6,13 @@
 #include <fenv.h>
 #endif
 
+#if defined(sun) && defined(__SVR4) && !defined(_LP64)
+#define USE_EXTENDED_FILE_STDIO 1
+#include <stdio.h>
+#include <stdio_ext.h>
+#include <signal.h>
+#endif
+
 int
 main(int argc, char **argv)
 {
@@ -17,5 +24,18 @@ main(int argc, char **argv)
 #ifdef __FreeBSD__
 	fedisableexcept(FE_OVERFLOW);
 #endif
+#ifdef USE_EXTENDED_FILE_STDIO
+	/*
+	 * enable extended FILE facility on OmniOS so that Python
+	 * apps can keep more than 256 file descriptors open
+	 */
+	struct rlimit rlp;
+	(void) getrlimit(RLIMIT_NOFILE, &rlp);
+	rlp.rlim_cur = rlp.rlim_max;
+	if (setrlimit(RLIMIT_NOFILE, &rlp) != -1) {
+		enable_extended_FILE_stdio(-1, 0);
+	}
+#endif
+
 	return Py_Main(argc, argv);
 }
