diff --git Python-2.6.4/Include/pyport.h Python-2.6.4/Include/pyport.h
diff -pruN '--exclude=*.orig' Python-2.7.14~/Include/pyport.h Python-2.7.14/Include/pyport.h
--- Python-2.7.14~/Include/pyport.h	2017-09-16 17:38:35.000000000 +0000
+++ Python-2.7.14/Include/pyport.h	2017-09-23 22:33:32.666006639 +0000
@@ -640,11 +640,6 @@ Please be conservative with adding new o
 in platform-specific #ifdefs.
 **************************************************************************/
 
-#ifdef SOLARIS
-/* Unchecked */
-extern int gethostname(char *, int);
-#endif
-
 #ifdef __BEOS__
 /* Unchecked */
 /* It's in the libs, but not the headers... - [cjh] */
