
The ERR_GET_FUNC macro was removed in OpenSSL 3.0

diff -wpruN '--exclude=*.orig' a~/buckets/ssl_buckets.c a/buckets/ssl_buckets.c
--- a~/buckets/ssl_buckets.c	1970-01-01 00:00:00
+++ a/buckets/ssl_buckets.c	1970-01-01 00:00:00
@@ -1326,7 +1326,11 @@ static int ssl_need_client_cert(SSL *ssl
             }
             else {
                 printf("OpenSSL cert error: %d %d %d\n", ERR_GET_LIB(err),
+#ifdef ERR_GET_FUNC
                        ERR_GET_FUNC(err),
+#else
+                       0,
+#endif
                        ERR_GET_REASON(err));
                 PKCS12_free(p12);
                 bio_meth_free(biom);
