diff -wpruN '--exclude=*.orig' a~/ext/openssl/ossl_pkey_rsa.c a/ext/openssl/ossl_pkey_rsa.c
--- a~/ext/openssl/ossl_pkey_rsa.c	1970-01-01 00:00:00
+++ a/ext/openssl/ossl_pkey_rsa.c	1970-01-01 00:00:00
@@ -939,7 +939,9 @@ Init_ossl_rsa(void)
     rb_define_method(cRSA, "params", ossl_rsa_get_params, 0);
 
     DefRSAConst(PKCS1_PADDING);
+#ifdef RSA_SSLV23_PADDING
     DefRSAConst(SSLV23_PADDING);
+#endif
     DefRSAConst(NO_PADDING);
     DefRSAConst(PKCS1_OAEP_PADDING);
 
