From 9c166b839b443ee67ecda9ca43c0ebb157880647 Mon Sep 17 00:00:00 2001
From: "Joshua M. Clulow" <josh@sysmgr.org>
Date: Thu, 12 Aug 2021 10:50:01 -0700
Subject: [PATCH] add cfmakeraw() for illumos

diff -wpruN --no-dereference '--exclude=*.orig' a~/term.c a/term.c
--- a~/term.c	1970-01-01 00:00:00
+++ a/term.c	1970-01-01 00:00:00
@@ -759,6 +759,22 @@ term_apply (int fd, int now)
 
 /***************************************************************************/
 
+#ifdef __illumos__
+void
+cfmakeraw(struct termios *tio)
+{
+    tio->c_iflag &= ~(IMAXBEL | IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR |
+        IGNCR | ICRNL | IXON);
+    tio->c_oflag &= ~OPOST;
+    tio->c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
+    tio->c_cflag &= ~(CSIZE | PARENB);
+    tio->c_cflag |= CS8;
+
+    tio->c_cc[VMIN] = 1;
+    tio->c_cc[VTIME] = 0;
+}
+#endif
+
 int
 term_set_raw (int fd)
 {
