diff -wpruN --no-dereference '--exclude=*.orig' a~/texk/kpathsea/texmf.cnf a/texk/kpathsea/texmf.cnf
--- a~/texk/kpathsea/texmf.cnf	1970-01-01 00:00:00
+++ a/texk/kpathsea/texmf.cnf	1970-01-01 00:00:00
@@ -59,7 +59,7 @@
 % SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
 % SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
 % Sorry for the off-by-one-generation names.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = $SELFAUTODIR/share
 
 % The main tree of distributed packages and programs:
 TEXMFDIST = $TEXMFROOT/texmf-dist
