diff -wpruN '--exclude=*.orig' texlive-20180414-source~/texk/kpathsea/texmf.cnf texlive-20180414-source/texk/kpathsea/texmf.cnf
--- texlive-20180414-source~/texk/kpathsea/texmf.cnf	2018-03-26 14:23:43.000000000 +0000
+++ texlive-20180414-source/texk/kpathsea/texmf.cnf	2018-10-14 17:51:56.208543279 +0000
@@ -58,7 +58,7 @@
 % SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
 % SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
 % Sorry for the off-by-one-generation names.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = $SELFAUTODIR/share
 
 % The main tree of distributed packages and programs:
 TEXMFDIST = $TEXMFROOT/texmf-dist
