diff -wpruN '--exclude=*.orig' a~/solaris/SolarisProcessList.h a/solaris/SolarisProcessList.h
--- a~/solaris/SolarisProcessList.h	1970-01-01 00:00:00
+++ a/solaris/SolarisProcessList.h	1970-01-01 00:00:00
@@ -22,7 +22,7 @@ in the source distribution for its full
 #include <sys/swap.h>
 
 #define ZONE_ERRMSGLEN 1024
-char zone_errmsg[ZONE_ERRMSGLEN];
+extern char zone_errmsg[ZONE_ERRMSGLEN];
 
 typedef struct CPUData_ {
    double userPercent;
