diff -wpruN '--exclude=*.orig' a~/CRT.h a/CRT.h
--- a~/CRT.h	1970-01-01 00:00:00
+++ a/CRT.h	1970-01-01 00:00:00
@@ -140,7 +140,7 @@ extern const char **CRT_treeStr;
 
 extern int CRT_delay;
 
-int* CRT_colors;
+extern int* CRT_colors;
 
 extern int CRT_colorSchemes[LAST_COLORSCHEME][LAST_COLORELEMENT];
 
@@ -150,13 +150,13 @@ extern int CRT_scrollHAmount;
 
 extern int CRT_scrollWheelVAmount;
 
-char* CRT_termType;
+extern char* CRT_termType;
 
 // TODO move color scheme to Settings, perhaps?
 
 extern int CRT_colorScheme;
 
-void *backtraceArray[128];
+extern void *backtraceArray[128];
 
 #if HAVE_SETUID_ENABLED
 
diff -wpruN '--exclude=*.orig' a~/solaris/SolarisProcessList.h a/solaris/SolarisProcessList.h
--- a~/solaris/SolarisProcessList.h	1970-01-01 00:00:00
+++ a/solaris/SolarisProcessList.h	1970-01-01 00:00:00
@@ -22,7 +22,7 @@ in the source distribution for its full
 #include <sys/swap.h>
 
 #define ZONE_ERRMSGLEN 1024
-char zone_errmsg[ZONE_ERRMSGLEN];
+extern char zone_errmsg[ZONE_ERRMSGLEN];
 
 typedef struct CPUData_ {
    double userPercent;
