diff -wpruN --no-dereference '--exclude=*.orig' a~/libav.h a/libav.h
--- a~/libav.h	1970-01-01 00:00:00
+++ a/libav.h	1970-01-01 00:00:00
@@ -174,7 +174,7 @@ lav_get_interlaced(AVStream *s)
 #define lav_codec_tag(s) s->codecpar->codec_tag
 #define lav_sample_rate(s) s->codecpar->sample_rate
 #define lav_bit_rate(s) s->codecpar->bit_rate
-#define lav_channels(s) s->codecpar->channels
+#define lav_channels(s) s->codecpar->ch_layout.nb_channels
 #define lav_width(s) s->codecpar->width
 #define lav_height(s) s->codecpar->height
 #define lav_profile(s) s->codecpar->profile
@@ -186,7 +186,7 @@ lav_get_interlaced(AVStream *s)
 #define lav_codec_tag(s) s->codec->codec_tag
 #define lav_sample_rate(s) s->codec->sample_rate
 #define lav_bit_rate(s) s->codec->bit_rate
-#define lav_channels(s) s->codec->channels
+#define lav_channels(s) s->codec->ch_layout.nb_channels
 #define lav_width(s) s->codec->width
 #define lav_height(s) s->codec->height
 #define lav_profile(s) s->codec->profile
