diff -wpruN '--exclude=*.orig' a~/minidlna.conf a/minidlna.conf
--- a~/minidlna.conf	1970-01-01 00:00:00
+++ a/minidlna.conf	1970-01-01 00:00:00
@@ -40,7 +40,7 @@ album_art_names=Cover.jpg/cover.jpg/Albu
 
 # set this to no to disable inotify monitoring to automatically discover new files
 # note: the default is yes
-inotify=yes
+inotify=no
 
 # set this to yes to enable support for streaming .jpg and .mp3 files to a TiVo supporting HMO
 enable_tivo=no
