diff -wpruN '--exclude=*.orig' a~/image_utils.c a/image_utils.c
--- a~/image_utils.c	1970-01-01 00:00:00
+++ a/image_utils.c	1970-01-01 00:00:00
@@ -190,7 +190,7 @@ jpeg_memory_src(j_decompress_ptr cinfo,
 	src->pub.bytes_in_buffer = bufsize;
 }
 
-jmp_buf setjmp_buffer;
+static jmp_buf setjmp_buffer;
 /* Don't exit on error like libjpeg likes to do */
 static void
 libjpeg_error_handler(j_common_ptr cinfo)
diff -wpruN '--exclude=*.orig' a~/metadata.c a/metadata.c
--- a~/metadata.c	1970-01-01 00:00:00
+++ a/metadata.c	1970-01-01 00:00:00
@@ -484,7 +484,7 @@ GetAudioMetadata(const char *path, const
 }
 
 /* For libjpeg error handling */
-jmp_buf setjmp_buffer;
+static jmp_buf setjmp_buffer;
 static void
 libjpeg_error_handler(j_common_ptr cinfo)
 {
