diff -wpruN --no-dereference '--exclude=*.orig' a~/CMakeLists.txt a/CMakeLists.txt
--- a~/CMakeLists.txt	1970-01-01 00:00:00
+++ a/CMakeLists.txt	1970-01-01 00:00:00
@@ -978,6 +978,10 @@ if(MINGW AND "${ZIG_HOST_TARGET_ARCH}" S
 endif()
 
 
+if(ZIG_BUILD_VERBOSE_LINK)
+  list(APPEND ZIG_BUILD_ARGS "--verbose-link")
+endif()
+
 add_custom_target(stage3 ALL
   DEPENDS "${PROJECT_BINARY_DIR}/stage3/bin/zig"
 )
