diff -wpruN --no-dereference '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -5351,7 +5351,7 @@ case $target_os in
         ;;
     sunos)
         SHFLAGS='-shared -Wl,-h,$$(@F)'
-        enabled x86 && append SHFLAGS -mimpure-text
+        [ "$cc_type" != clang ] && enabled x86 && append SHFLAGS -mimpure-text
         network_extralibs="-lsocket -lnsl"
         add_cppflags -D__EXTENSIONS__
         # When using suncc to build, the Solaris linker will mark
