see: https://www.illumos.org/issues/14418

diff -wpruN '--exclude=*.orig' a~/deps/v8/src/base/platform/platform-posix.cc a/deps/v8/src/base/platform/platform-posix.cc
--- a~/deps/v8/src/base/platform/platform-posix.cc	1970-01-01 00:00:00
+++ a/deps/v8/src/base/platform/platform-posix.cc	1970-01-01 00:00:00
@@ -72,14 +72,6 @@
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
-#if defined(V8_OS_SOLARIS)
-#if (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE > 2) || defined(__EXTENSIONS__)
-extern "C" int madvise(caddr_t, size_t, int);
-#else
-extern int madvise(caddr_t, size_t, int);
-#endif
-#endif
-
 #ifndef MADV_FREE
 #define MADV_FREE MADV_DONTNEED
 #endif
