diff -wpruN '--exclude=*.orig' a~/api4/websocket.go a/api4/websocket.go
--- a~/api4/websocket.go	1970-01-01 00:00:00
+++ a/api4/websocket.go	1970-01-01 00:00:00
@@ -49,7 +49,7 @@ func connectWebSocket(c *Context, w http
 		c.App.HubRegister(wc)
 	}
 
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == "windows" || runtime.GOOS == "illumos" {
 		wc.BlockingPump()
 	} else {
 		go wc.Pump()
diff -wpruN '--exclude=*.orig' a~/app/server.go a/app/server.go
--- a~/app/server.go	1970-01-01 00:00:00
+++ a/app/server.go	1970-01-01 00:00:00
@@ -231,7 +231,7 @@ func NewServer(options ...Option) (*Serv
 	}
 
 	// epoll/kqueue is not available on Windows.
-	if runtime.GOOS != "windows" {
+	if runtime.GOOS != "windows" && runtime.GOOS != "illumos" {
 		poller, err := netpoll.New(nil)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to create a netpoll instance")
diff -wpruN '--exclude=*.orig' a~/app/web_conn.go a/app/web_conn.go
--- a~/app/web_conn.go	1970-01-01 00:00:00
+++ a/app/web_conn.go	1970-01-01 00:00:00
@@ -77,7 +77,7 @@ func (a *App) NewWebConn(ws net.Conn, se
 		UserId:             session.UserId,
 		T:                  t,
 		Locale:             locale,
-		isWindows:          runtime.GOOS == "windows",
+		isWindows:          runtime.GOOS == "windows" || runtime.GOOS == "illumos",
 		endWritePump:       make(chan struct{}),
 		pumpFinished:       make(chan struct{}),
 	}
