diff -wpruN '--exclude=*.orig' a~/build/release.mk a/build/release.mk
--- a~/build/release.mk	1970-01-01 00:00:00
+++ a/build/release.mk	1970-01-01 00:00:00
@@ -1,5 +1,9 @@
 dist: | check-style test package
 
+build-illumos:
+	@echo Build illumos amd64
+	env GOOS=illumos GOARCH=amd64 $(GO) build -o $(GOBIN) $(GOFLAGS) -trimpath -ldflags '$(LDFLAGS)' ./...
+
 build-linux:
 	@echo Build Linux amd64
 ifeq ($(BUILDER_GOOS_GOARCH),"linux_amd64")
@@ -134,18 +138,6 @@ endif
 		cp ../manifest.txt $(DIST_PATH); \
 	fi
 
-	@# Import Mattermost plugin public key
-	gpg --import build/plugin-production-public-key.gpg
-
-	@# Download prepackaged plugins
-	mkdir -p tmpprepackaged
-	@cd tmpprepackaged && for plugin_package in $(PLUGIN_PACKAGES) ; do \
-		for ARCH in "osx-amd64" "windows-amd64" "linux-amd64" ; do \
-			curl -f -O -L https://plugins-store.test.mattermost.com/release/$$plugin_package-$$ARCH.tar.gz; \
-			curl -f -O -L https://plugins-store.test.mattermost.com/release/$$plugin_package-$$ARCH.tar.gz.sig; \
-		done; \
-	done
-
 package-general:
 	@# Create needed directories
 	mkdir -p $(DIST_PATH_GENERIC)/bin
@@ -190,6 +182,15 @@ else
 	done
 endif
 
+package-illumos: package-prep
+	@# Create needed directories
+	mkdir -p $(DIST_PATH)/bin
+	mkdir -p $(DIST_PATH)/logs
+	mkdir -p $(DIST_PATH)/prepackaged_plugins
+
+	@# Copy binary
+	cp $(GOBIN)/mattermost $(DIST_PATH)/bin # from native bin dir, not cross-compiled
+
 package-osx-amd64: package-prep
 	DIST_PATH_GENERIC=$(DIST_PATH_OSX_AMD64) CURRENT_PACKAGE_ARCH=darwin_amd64 PLUGIN_ARCH=osx-amd64 MMCTL_PLATFORM="Darwin-x86_64" MM_BIN_NAME=mattermost $(MAKE) package-general
 	@# Package
