--- open-vm-tools-10.1.10-6082533/lib/nicInfo/nicInfoPosix.c	2017-07-28 22:19:20.000000000 +0000
+++ l/lib/nicInfo/nicInfoPosix.c	2017-09-18 19:31:27.588587527 +0000
@@ -34,11 +34,8 @@
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <errno.h>
-#if defined(__FreeBSD__) || defined(__APPLE__)
-# include <sys/sysctl.h>
 # include <ifaddrs.h>
 # include <net/if.h>
-#endif
 #ifndef NO_DNET
 # ifdef DNET_IS_DUMBNET
 #  include <dumbnet.h>
@@ -168,7 +168,7 @@
 }
 
 
-#ifdef USERWORLD
+#if defined(USERWORLD) || defined(sun)
 /*
  ******************************************************************************
  * CountNetmaskBits --                                                   */ /**
@@ -256,7 +256,7 @@
    }
 
    return TRUE;
-#elif defined(USERWORLD)
+#elif defined(USERWORLD) || defined(sun)
    struct ifaddrs *ifaddrs = NULL;
 
    if (getifaddrs(&ifaddrs) == 0 && ifaddrs != NULL) {
@@ -348,7 +345,7 @@
  *
  ******************************************************************************
  */
-#if defined(__FreeBSD__) || defined(__APPLE__) || defined(USERWORLD)
+#if 1
 
 char *
 GuestInfoGetPrimaryIP(void)
