diff -pruN '--exclude=*.orig' open-vm-tools-10.2.0-7253323~/lib/nicInfo/nicInfoPosix.c open-vm-tools-10.2.0-7253323/lib/nicInfo/nicInfoPosix.c
--- open-vm-tools-10.2.0-7253323~/lib/nicInfo/nicInfoPosix.c	2017-12-15 19:09:51.000000000 +0000
+++ open-vm-tools-10.2.0-7253323/lib/nicInfo/nicInfoPosix.c	2017-12-20 14:32:35.051752694 +0000
@@ -34,11 +34,8 @@
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <errno.h>
-#if defined(__FreeBSD__) || defined(__APPLE__)
-# include <sys/sysctl.h>
 # include <ifaddrs.h>
 # include <net/if.h>
-#endif
 #ifndef NO_DNET
 # ifdef DNET_IS_DUMBNET
 #  include <dumbnet.h>
@@ -186,7 +183,7 @@ GuestInfoGetFqdn(int outBufLen,    // IN
 }
 
 
-#if defined(USERWORLD) || defined(USE_SLASH_PROC) || (defined(__linux__) && defined(NO_DNET))
+#if defined(sun) || defined(USERWORLD) || defined(USE_SLASH_PROC) || (defined(__linux__) && defined(NO_DNET))
 /*
  ******************************************************************************
  * CountNetmaskBits --                                                   */ /**
@@ -225,7 +222,7 @@ CountNetmaskBitsV4(struct sockaddr *netm
 }
 #endif
 
-#if defined(USERWORLD) || (defined(__linux__) && defined(NO_DNET))
+#if defined(USERWORLD) || defined(sun) || (defined(__linux__) && defined(NO_DNET))
 static unsigned
 CountNetmaskBitsV6(struct sockaddr *netmask)
 {
@@ -379,7 +376,7 @@ Bool GuestInfoGetNicInfo(NicInfoV3 *nicI
    }
 
    return TRUE;
-#elif defined(USERWORLD) || defined(__linux__)
+#elif defined(USERWORLD) || defined(__linux__) || defined(sun)
    struct ifaddrs *ifaddrs = NULL;
 
    if (getifaddrs(&ifaddrs) == 0 && ifaddrs != NULL) {
@@ -425,6 +422,7 @@ Bool GuestInfoGetNicInfo(NicInfoV3 *nicI
  */
 #if defined(__FreeBSD__) || \
     defined(__APPLE__) || \
+    defined(sun) || \
     defined(USERWORLD) || \
     (defined(__linux__) && defined(NO_DNET))
 
