diff -pruN '--exclude=*.orig' open-vm-tools-10.1.15~/services/vmtoolsd/mainLoop.c open-vm-tools-10.1.15/services/vmtoolsd/mainLoop.c
--- open-vm-tools-10.1.15~/services/vmtoolsd/mainLoop.c	2017-09-29 21:15:57.000000000 +0000
+++ open-vm-tools-10.1.15/services/vmtoolsd/mainLoop.c	2017-10-03 09:31:21.147907737 +0000
@@ -71,7 +71,15 @@ ToolsCoreCleanup(ToolsServiceState *stat
       state->ctx.rpc = NULL;
    }
    g_key_file_free(state->ctx.config);
-   g_main_loop_unref(state->ctx.mainLoop);
+/*
+ * (NOTE:  Bug IDs here are from Delphix, not OmniTI.)
+ * Cleaning up is good, but not necessary, and this currently
+ * causes a core dump when vmtoolsd dies because of a known bug in
+ * glib; see Bug 28773. As a workaround, let the kernel free the
+ * memory for us, as it always will when the process exits. This
+ * should be removed when glib is fixed (see bug 33706).
+ */
+ /* g_main_loop_unref(state->ctx.mainLoop); */
 
 #if defined(G_PLATFORM_WIN32)
    if (state->ctx.comInitialized) {
