diff -pruN '--exclude=*.orig' open-vm-tools-10.1.15~/lib/include/sigPosixRegs.h open-vm-tools-10.1.15/lib/include/sigPosixRegs.h
--- open-vm-tools-10.1.15~/lib/include/sigPosixRegs.h	2017-09-29 21:15:57.000000000 +0000
+++ open-vm-tools-10.1.15/lib/include/sigPosixRegs.h	2017-10-03 09:31:21.409516608 +0000
@@ -70,7 +70,7 @@
 
 #include <signal.h>
 #ifndef __ANDROID__
-#include <sys/ucontext.h>
+#include <ucontext.h>
 #endif
 
 #if __linux__
