--- sudo-1.8.20p2/plugins/sudoers/bsm_audit.c	Sun Jul  9 12:35:16 2017
+++ sudo-1.8.20p2/plugins/sudoers/bsm_audit.c	Sun Jul  9 12:34:08 2017
@@ -34,6 +34,7 @@

 #include "sudoers.h"
 #include "bsm_audit.h"
+#define  AUE_sudo 6650

 /*
  * Solaris auditon() returns EINVAL if BSM audit not configured.
@@ -163,11 +164,7 @@
 		debug_return_int(-1);
 	}
 	au_write(aufd, tok);
-#ifdef __sun
-	if (au_close(aufd, 1, sudo_audit_event, 0) == -1)
-#else
 	if (au_close(aufd, 1, sudo_audit_event) == -1)
-#endif
 	{
 		sudo_warn(U_("unable to commit audit record"));
 		debug_return_int(-1);
@@ -251,11 +248,7 @@
 		debug_return_int(-1);
 	}
 	au_write(aufd, tok);
-#ifdef __sun
-	if (au_close(aufd, 1, sudo_audit_event, PAD_FAILURE) == -1)
-#else
 	if (au_close(aufd, 1, sudo_audit_event) == -1)
-#endif
 	{
 		sudo_warn(U_("unable to commit audit record"));
 		debug_return_int(-1);
