diff -pruN '--exclude=*.orig' sudo-1.8.21p2~/Makefile.in sudo-1.8.21p2/Makefile.in
--- sudo-1.8.21p2~/Makefile.in	2017-08-23 20:07:28.000000000 +0000
+++ sudo-1.8.21p2/Makefile.in	2017-12-03 17:16:12.659692940 +0000
@@ -63,7 +63,6 @@ SHELL = @SHELL@
 SED = @SED@
 
 INSTALL = $(SHELL) $(top_srcdir)/install-sh -c
-INSTALL_OWNER = -o $(install_uid) -g $(install_gid)
 
 ECHO_N = @ECHO_N@
 ECHO_C = @ECHO_C@
