diff -pruN '--exclude=*.orig' sudo-1.8.21p2~/include/sudo_debug.h sudo-1.8.21p2/include/sudo_debug.h
--- sudo-1.8.21p2~/include/sudo_debug.h	2017-05-10 17:38:43.000000000 +0000
+++ sudo-1.8.21p2/include/sudo_debug.h	2017-12-03 18:49:50.967208196 +0000
@@ -17,6 +17,7 @@
 #ifndef SUDO_DEBUG_H
 #define SUDO_DEBUG_H
 
+#include "sudo_compat.h"
 #include <stdarg.h>
 #ifdef HAVE_STDBOOL_H
 # include <stdbool.h>
