diff -wpruN '--exclude=*.orig' a~/bin/acmefetch a/bin/acmefetch
--- a~/bin/acmefetch	1970-01-01 00:00:00
+++ a/bin/acmefetch	1970-01-01 00:00:00
@@ -36,7 +36,7 @@ sub main()
     if($opt{man})      { pod2usage(-exitstatus => 0, -verbose => 2) }
     if($opt{version})  { print "acmefetch $VERSION\n"; exit(0) }
 
-    my $cfg = loadCfg($opt{cfg} // $FindBin::RealBin.'/../etc/acmefetch.cfg');
+    my $cfg = loadCfg($opt{cfg} // '/etc/opt/ooce/acmefetch/acmefetch.cfg');
 
     $cfg->{GENERAL}{host} = $opt{staging} ? $cfg->{GENERAL}->{ACMEstaging}
                                             : $cfg->{GENERAL}->{ACMEservice};
