diff -wpruN '--exclude=*.orig' a~/lib/Driver/ToolChains/Solaris.cpp a/lib/Driver/ToolChains/Solaris.cpp
--- a~/lib/Driver/ToolChains/Solaris.cpp	1970-01-01 00:00:00
+++ a/lib/Driver/ToolChains/Solaris.cpp	1970-01-01 00:00:00
@@ -97,12 +97,14 @@ void solaris::Linker::ConstructJob(Compi
         Args.MakeArgString(getToolChain().GetFilePath("crtbegin.o")));
   }
 
+#if 0
   // Provide __start___sancov_guards.  Solaris ld doesn't automatically create
   // __start_SECNAME labels.
   CmdArgs.push_back("--whole-archive");
   CmdArgs.push_back(
       getToolChain().getCompilerRTArgString(Args, "sancov_begin", false));
   CmdArgs.push_back("--no-whole-archive");
+#endif
 
   getToolChain().AddFilePathLibArgs(Args, CmdArgs);
 
@@ -132,12 +134,14 @@ void solaris::Linker::ConstructJob(Compi
       linkSanitizerRuntimeDeps(getToolChain(), CmdArgs);
   }
 
+#if 0
   // Provide __stop___sancov_guards.  Solaris ld doesn't automatically create
   // __stop_SECNAME labels.
   CmdArgs.push_back("--whole-archive");
   CmdArgs.push_back(
       getToolChain().getCompilerRTArgString(Args, "sancov_end", false));
   CmdArgs.push_back("--no-whole-archive");
+#endif
 
   if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles)) {
     CmdArgs.push_back(
