diff -wpruN --no-dereference '--exclude=*.orig' a~/src/builtins/ulimit.rs a/src/builtins/ulimit.rs
--- a~/src/builtins/ulimit.rs	1970-01-01 00:00:00
+++ a/src/builtins/ulimit.rs	1970-01-01 00:00:00
@@ -20,7 +20,7 @@ pub mod limits {
         pub const DATA: libc::c_int = libc::RLIMIT_DATA as _;
         pub const FSIZE: libc::c_int = libc::RLIMIT_FSIZE as _;
         cfg_if!(
-            if #[cfg(cygwin)] {
+            if #[cfg(any(cygwin,illumos))] {
                 pub const MEMLOCK: libc::c_int = -1;
             } else {
                 pub const MEMLOCK: libc::c_int = libc::RLIMIT_MEMLOCK as _;
@@ -30,7 +30,7 @@ pub mod limits {
         pub const STACK: libc::c_int = libc::RLIMIT_STACK as _;
         pub const CPU: libc::c_int = libc::RLIMIT_CPU as _;
         cfg_if!(
-            if #[cfg(cygwin)] {
+            if #[cfg(any(cygwin,illumos))] {
                 pub const NPROC: libc::c_int = -1;
             } else {
                 pub const NPROC: libc::c_int = libc::RLIMIT_NPROC as _;
