diff -wpruN --no-dereference '--exclude=*.orig' a~/Cargo.toml a/Cargo.toml
--- a~/Cargo.toml	1970-01-01 00:00:00
+++ a/Cargo.toml	1970-01-01 00:00:00
@@ -93,3 +93,7 @@ use-jemalloc = ["jemallocator"]
 completions = ["clap_complete"]
 base = ["use-jemalloc"]
 default = ["use-jemalloc", "completions"]
+
+[patch.crates-io]
+users = { git = "https://github.com/papertigers/rust-users", branch = "illumos" }
+argmax = { git = "https://github.com/omniosorg/argmax", branch = "illumos" }
