diff -wpruN '--exclude=*.orig' a~/Cargo.toml a/Cargo.toml
--- a~/Cargo.toml	1970-01-01 00:00:00
+++ a/Cargo.toml	1970-01-01 00:00:00
@@ -61,7 +61,7 @@ libc = "0.2"
 # FIXME: Re-enable jemalloc on macOS
 # jemalloc is currently disabled on macOS due to a bug in jemalloc in combination with macOS
 # Catalina. See https://github.com/sharkdp/fd/issues/498 for details.
-[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_env = "musl")))'.dependencies]
+[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_env = "musl"), not(target_os = "illumos")))'.dependencies]
 jemallocator = "0.3.0"
 
 [dev-dependencies]
diff -wpruN '--exclude=*.orig' a~/src/main.rs a/src/main.rs
--- a~/src/main.rs	1970-01-01 00:00:00
+++ a/src/main.rs	1970-01-01 00:00:00
@@ -38,7 +38,8 @@ use crate::regex_helper::{pattern_has_up
     not(windows),
     not(target_os = "android"),
     not(target_os = "macos"),
-    not(target_env = "musl")
+    not(target_env = "musl"),
+    not(target_os = "illumos")
 ))]
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
