diff -wpruN --no-dereference '--exclude=*.orig' a~/plugins-root/check_dhcp.c a/plugins-root/check_dhcp.c
--- a~/plugins-root/check_dhcp.c	1970-01-01 00:00:00
+++ a/plugins-root/check_dhcp.c	1970-01-01 00:00:00
@@ -82,7 +82,7 @@ const char *email = "devel@nagios-plugin
 #include <signal.h>
 #include <sys/dlpi.h>
 #include <sys/poll.h>
-#include <sys/stropts.h>
+#include <stropts.h>
 
 #define bcopy(source, destination, length) memcpy(destination, source, length)
 
