diff -pruN '--exclude=*.orig' p7zip_16.02~/makefile.common p7zip_16.02/makefile.common
--- p7zip_16.02~/makefile.common	2016-05-19 10:20:02.000000000 +0000
+++ p7zip_16.02/makefile.common	2018-05-02 12:40:09.737857809 +0000
@@ -1,10 +1,10 @@
 MKDIR = mkdir -p
 
-DEST_HOME=/usr/local
-DEST_BIN=$(DEST_HOME)/bin
-DEST_SHARE=$(DEST_HOME)/lib/p7zip
-DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
-DEST_MAN=$(DEST_HOME)/man
+#DEST_HOME=/usr/local
+#DEST_BIN=$(DEST_HOME)/bin
+#DEST_SHARE=$(DEST_HOME)/lib/p7zip
+#DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
+#DEST_MAN=$(DEST_HOME)/man
 
 .PHONY: default all all2 7za 7zG 7zFM sfx 7z 7zr Client7z common common7z LangAndHelp tar_bin depend test test_7z test_7zr test_7zG test_Client7z LzmaCon test_LzmaCon  all_test app cppcheck lzham test_lib other
 
