diff -wpruN --no-dereference '--exclude=*.orig' a~/CMakeLists.txt a/CMakeLists.txt
--- a~/CMakeLists.txt	1970-01-01 00:00:00
+++ a/CMakeLists.txt	1970-01-01 00:00:00
@@ -1256,6 +1256,8 @@ ${_}            pdcurses        ${e}${CU
 ${_}                            ${e}${CURSES_LIBRARIES}")
 endif()
 
+list(APPEND CMAKE_INSTALL_RPATH "/opt/ooce/lib/amd64")
+
 if(C_LINUX)
 if(SYSTEMD_PROGRAM_FOUND)
 message("\
