diff -wpruN '--exclude=*.orig' a~/Makefile a/Makefile
--- a~/Makefile	1970-01-01 00:00:00
+++ a/Makefile	1970-01-01 00:00:00
@@ -83,7 +83,7 @@ verify-healing:
 # Builds minio locally.
 build: checks
 	@echo "Building minio binary to './minio'"
-	@GO111MODULE=on CGO_ENABLED=0 go build -tags kqueue -trimpath --ldflags "$(LDFLAGS)" -o $(PWD)/minio 1>/dev/null
+	@GO111MODULE=on CGO_ENABLED=1 go build -tags kqueue -trimpath --ldflags "$(LDFLAGS)" -o $(PWD)/minio 1>/dev/null
 
 docker: build
 	@docker build -t $(TAG) . -f Dockerfile.dev
