diff -wpruN --no-dereference '--exclude=*.orig' a~/_deps/pkg/mod/github.com/tinylib/msgp@v1.1.9/msgp/file.go a/_deps/pkg/mod/github.com/tinylib/msgp@v1.1.9/msgp/file.go
--- a~/_deps/pkg/mod/github.com/tinylib/msgp@v1.1.9/msgp/file.go	1970-01-01 00:00:00
+++ a/_deps/pkg/mod/github.com/tinylib/msgp@v1.1.9/msgp/file.go	1970-01-01 00:00:00
@@ -1,5 +1,5 @@
-//go:build (linux || darwin || dragonfly || freebsd || netbsd || openbsd) && !appengine && !tinygo
-// +build linux darwin dragonfly freebsd netbsd openbsd
+//go:build (linux || darwin || dragonfly || freebsd || netbsd || openbsd || illumos) && !appengine && !tinygo
+// +build linux darwin dragonfly freebsd netbsd openbsd illumos
 // +build !appengine
 // +build !tinygo
 
