diff -wpruN '--exclude=*.orig' a~/config/config.go a/config/config.go
--- a~/config/config.go	1970-01-01 00:00:00
+++ a/config/config.go	1970-01-01 00:00:00
@@ -671,8 +671,8 @@ func (t *HookEnum) UnmarshalYAML(u func(
 }
 
 var ConfigFileDefaultLocations = []string{
+	"/etc/opt/ooce/zrepl/zrepl.yml",
 	"/etc/zrepl/zrepl.yml",
-	"/usr/local/etc/zrepl/zrepl.yml",
 }
 
 func ParseConfig(path string) (i *Config, err error) {
