From 08bcbbb75394b80bf84b7f77428a84ef1bac48ed Mon Sep 17 00:00:00 2001
From: Ryan Zezeski <ryan@zinascii.com>
Date: Wed, 3 Sep 2025 17:11:30 -0400
Subject: [PATCH] keep frame pointers

diff -wpruN --no-dereference '--exclude=*.orig' a~/build.zig a/build.zig
--- a~/build.zig	1970-01-01 00:00:00
+++ a/build.zig	1970-01-01 00:00:00
@@ -203,6 +203,11 @@ pub fn build(b: *std.Build) !void {
     exe.pie = pie;
     exe.entitlements = entitlements;
 
+    //
+    // We prefer to have the frame pointer in illumos.
+    //
+    exe.root_module.omit_frame_pointer = false;
+
     const use_llvm = b.option(bool, "use-llvm", "Use the llvm backend");
     exe.use_llvm = use_llvm;
     exe.use_lld = use_llvm;
