diff -wpruN '--exclude=*.orig' a~/doc/mutt.man a/doc/mutt.man
--- a~/doc/mutt.man	1970-01-01 00:00:00
+++ a/doc/mutt.man	1970-01-01 00:00:00
@@ -86,9 +86,6 @@ OpenPGP, and a threaded sorting mode.
 .BR Note :
 This manual page gives a brief overview of the \fBmutt\fP executable command
 line options.
-A copy of the full manual is located in \fI@docdir@\fP, in text, HTML, and/or
-PDF format.
-Please refer to the manual to learn how to use and configure Mutt.
 .SH OPTIONS
 .TP
 .BI \-A " alias"
@@ -277,9 +274,6 @@ System mapping between MIME types and fi
 .TP
 .I @bindir@/mutt_dotlock
 The privileged dotlocking program.
-.TP
-.I @docdir@/manual.txt
-The Mutt manual.
 .SH BUGS
 .PP
 None.  Mutts have fleas, not bugs.
