diff -wpruN --no-dereference '--exclude=*.orig' a~/Zend/zend_portability.h a/Zend/zend_portability.h
--- a~/Zend/zend_portability.h	1970-01-01 00:00:00
+++ a/Zend/zend_portability.h	1970-01-01 00:00:00
@@ -761,7 +761,9 @@ extern "C++" {
 /** @deprecated */
 #define ZEND_CGG_DIAGNOSTIC_IGNORED_END ZEND_DIAGNOSTIC_IGNORED_END
 
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) /* C11 */
+#if defined(__cplusplus) || (defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 202311L)) /* C++ or C23 */
+# define ZEND_STATIC_ASSERT(c, m) static_assert((c), m)
+#elif defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) /* C11 */
 # define ZEND_STATIC_ASSERT(c, m) _Static_assert((c), m)
 #else
 # define ZEND_STATIC_ASSERT(c, m)
