diff -pruN '--exclude=*.orig' unzip60~/unix/Makefile unzip60/unix/Makefile
--- unzip60~/unix/Makefile	2009-01-18 22:41:18.000000000 +0000
+++ unzip60/unix/Makefile	2018-03-29 23:42:33.883704643 +0000
@@ -121,9 +121,9 @@ INSTALL_PROGRAM = $(INSTALL)
 INSTALL_D = mkdir -p
 # on some systems, manext=l and MANDIR=/usr/man/man$(manext) may be appropriate
 manext = 1
-prefix = /usr/local
-BINDIR = $(prefix)/bin#			where to install executables
-MANDIR = $(prefix)/man/man$(manext)#	where to install man pages
+prefix = $(PREFIX)
+BINDIR = $(prefix)/bin/$(ISAPART)#			where to install executables
+MANDIR = $(prefix)/share/man/man$(manext)#	where to install man pages
 INSTALLEDBIN = $(BINDIR)/funzip$E $(BINDIR)/unzip$E $(BINDIR)/unzipsfx$E \
 	$(BINDIR)/zipgrep$E $(BINDIR)/zipinfo$E
 INSTALLEDMAN = $(MANDIR)/funzip.$(manext) $(MANDIR)/unzip.$(manext) \
