diff -wpruN --no-dereference '--exclude=*.orig' a~/imagemagick.m4 a/imagemagick.m4
--- a~/imagemagick.m4	1970-01-01 00:00:00
+++ a/imagemagick.m4	1970-01-01 00:00:00
@@ -123,7 +123,7 @@ AC_DEFUN([IM_FIND_IMAGEMAGICK],[
   AC_MSG_RESULT([found in $IM_WAND_BINARY])
 
 # This is used later for cflags and libs
-  export PKG_CONFIG_PATH="${IM_IMAGEMAGICK_PREFIX}/${PHP_LIBDIR}/pkgconfig"
+  export PKG_CONFIG_PATH="${IM_IMAGEMAGICK_PREFIX}/${PHP_LIBDIR}/amd64/pkgconfig"
   
 # Check version
 #
