diff -wpruN --no-dereference '--exclude=*.orig' a~/src/VBox/Runtime/r3/posix/RTTimeZoneGetCurrent-posix.cpp a/src/VBox/Runtime/r3/posix/RTTimeZoneGetCurrent-posix.cpp
--- a~/src/VBox/Runtime/r3/posix/RTTimeZoneGetCurrent-posix.cpp	1970-01-01 00:00:00
+++ a/src/VBox/Runtime/r3/posix/RTTimeZoneGetCurrent-posix.cpp	1970-01-01 00:00:00
@@ -51,7 +51,7 @@
 #include <iprt/symlink.h>
 #include <iprt/stream.h>
 
-#if defined(RT_OS_DARWIN) || defined(RT_OS_SOLARIS)
+#if defined(RT_OS_DARWIN) || (defined(RT_OS_SOLARIS) && !defined(__illumos__))
 # include <tzfile.h>
 #else
 # define TZDIR                   "/usr/share/zoneinfo"
