diff -wpruN '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -3794,13 +3794,6 @@ case "${target}" in
     noconfigdirs="$noconfigdirs target-newlib target-libgloss ${libgcj}"
     noconfigdirs="$noconfigdirs target-libiberty"
     noconfigdirs="$noconfigdirs target-libmudflap target-libgomp target-libitm"
-    case "${host}" in
-        aarch64*-*-solaris2*)
-        ;;
-        *)
-            noconfigdirs="$noconfigdirs target-libstdc++-v3"
-        ;;
-    esac
     ;;
   alpha*-dec-osf*)
     # ld works, but does not support shared libraries.
diff -wpruN '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -1069,13 +1069,6 @@ case "${target}" in
     noconfigdirs="$noconfigdirs target-newlib target-libgloss ${libgcj}"
     noconfigdirs="$noconfigdirs target-libiberty"
     noconfigdirs="$noconfigdirs target-libmudflap target-libgomp target-libitm"
-    case "${host}" in
-        aarch64*-*-solaris2*)
-        ;;
-        *)
-            noconfigdirs="$noconfigdirs target-libstdc++-v3"
-        ;;
-    esac
     ;;
   alpha*-dec-osf*)
     # ld works, but does not support shared libraries.
