diff -wpruN --no-dereference '--exclude=*.orig' a~/ext/libxml/libxml.c a/ext/libxml/libxml.c
--- a~/ext/libxml/libxml.c	1970-01-01 00:00:00
+++ a/ext/libxml/libxml.c	1970-01-01 00:00:00
@@ -736,7 +736,7 @@ PHP_LIBXML_API void php_libxml_ctx_warni
 	va_end(args);
 }
 
-PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, xmlErrorPtr error)
+PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, const xmlError *error)
 {
 	_php_list_set_error_structure(error, NULL);
 
