icu4c 75.1 uses C++17 features

diff -wpruN --no-dereference '--exclude=*.orig' a~/ext/intl/config.m4 a/ext/intl/config.m4
--- a~/ext/intl/config.m4	1970-01-01 00:00:00
+++ a/ext/intl/config.m4	1970-01-01 00:00:00
@@ -85,7 +85,7 @@ if test "$PHP_INTL" != "no"; then
     breakiterator/codepointiterator_methods.cpp"
 
   PHP_REQUIRE_CXX()
-  PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
+  PHP_CXX_COMPILE_STDCXX(17, mandatory, PHP_INTL_STDCXX)
   PHP_INTL_CXX_FLAGS="$INTL_COMMON_FLAGS $PHP_INTL_STDCXX $ICU_CXXFLAGS"
   case $host_alias in
   *cygwin*) PHP_INTL_CXX_FLAGS="$PHP_INTL_CXX_FLAGS -D_POSIX_C_SOURCE=200809L"
