diff -pruN '--exclude=*.orig' ior-3.1.0~/src/ior.h ior-3.1.0/src/ior.h
--- ior-3.1.0~/src/ior.h	2017-08-24 01:36:29.000000000 +0000
+++ ior-3.1.0/src/ior.h	2018-04-09 19:43:47.976491154 +0000
@@ -48,6 +48,10 @@ extern int tasksPerNode;
 extern int verbose;
 extern MPI_Comm testComm;
 
+#ifdef __sun
+#define MIN(x, y) ((x) < (y) ? (x) : (y))
+#define MAX(x, y) ((x) > (y) ? (x) : (y))
+#endif
 
 /******************** DATA Packet Type ***************************************/
 /* Holds the types of data packets: generic, offset, timestamp, incompressible */
