diff -wpruN --no-dereference '--exclude=*.orig' a~/server/build/release.mk a/server/build/release.mk
--- a~/server/build/release.mk	1970-01-01 00:00:00
+++ a/server/build/release.mk	1970-01-01 00:00:00
@@ -1,5 +1,9 @@
 dist: | check-style test package
 
+build-illumos:
+	@echo Build illumos amd64
+	env GOOS=illumos GOARCH=amd64 $(GO) build -o $(GOBIN) $(GOFLAGS) -trimpath -tags '$(BUILD_TAGS) production' -ldflags '$(LDFLAGS)' ./...
+
 build-linux: build-linux-amd64 build-linux-arm64
 
 build-linux-amd64:
