diff -wpruN '--exclude=*.orig' a/src/github.com/mattermost/mattermost-server~/build/release.mk a/src/github.com/mattermost/mattermost-server/build/release.mk
--- a~/build/release.mk	1970-01-01 00:00:00
+++ a/build/release.mk	1970-01-01 00:00:00
@@ -1,6 +1,10 @@
 
 dist: | check-style test package
 
+build-illumos:
+	@echo Build illumos amd64
+	env GOOS=solaris GOARCH=amd64 $(GO) install -i $(GOFLAGS) -ldflags '$(LDFLAGS)' ./...
+
 build-linux:
 	@echo Build Linux amd64
 	env GOOS=linux GOARCH=amd64 $(GO) install -i $(GOFLAGS) -ldflags '$(LDFLAGS)' ./...
@@ -71,6 +75,16 @@ endif
 
 	@# ----- PLATFORM SPECIFIC -----
 
+	@# Make illumos package
+	@# Copy binary
+ifeq ($(BUILDER_GOOS_GOARCH),"solaris_amd64")
+	cp $(GOPATH)/bin/mattermost $(DIST_PATH)/bin # from native bin dir, not cross-compiled
+	cp $(GOPATH)/bin/platform $(DIST_PATH)/bin # from native bin dir, not cross-compiled
+else
+	cp $(GOPATH)/bin/solaris_amd64/mattermost $(DIST_PATH)/bin # from cross-compiled bin dir
+	cp $(GOPATH)/bin/solaris_amd64/platform $(DIST_PATH)/bin # from cross-compiled bin dir
+endif
+
 	@# Make osx package
 	@# Copy binary
 ifeq ($(BUILDER_GOOS_GOARCH),"darwin_amd64")
