diff -wpruN '--exclude=*.orig' a~/pkg/disk/stat_openbsd.go a/pkg/disk/stat_openbsd.go
--- a~/pkg/disk/stat_openbsd.go	1970-01-01 00:00:00
+++ a/pkg/disk/stat_openbsd.go	1970-01-01 00:00:00
@@ -1,4 +1,4 @@
-// +build openbsd
+// +build openbsd illumos
 
 /*
  * MinIO Cloud Storage, (C) 2019-2020 MinIO, Inc.
@@ -22,15 +22,15 @@ import (
 	"os/user"
 	"strconv"
 	"strings"
-	"syscall"
+	"golang.org/x/sys/unix"
 )
 
 // GetFileSystemAttrs return the file system attribute as string; containing mode,
 // uid, gid, uname, Gname, atime, mtime, ctime and md5
 func GetFileSystemAttrs(file string) (string, error) {
 
-	st := syscall.Stat_t{}
-	err := syscall.Stat(file, &st)
+	st := unix.Stat_t{}
+	err := unix.Stat(file, &st)
 	if err != nil {
 		return "", err
 	}
