From 1f941132971b867847159f7d8f20be45ab5956c2 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <illumos@fiddaman.net>
Date: Mon, 28 Nov 2022 14:53:22 +0000
Subject: Do not pass --dynamic-list to linker

---
 plugins/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plugins/meson.build b/plugins/meson.build
index 51b4350c2a..e74464f9b8 100644
--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -8,8 +8,6 @@ if not enable_modules
       capture: true,
       command: ['sed', '-ne', 's/^[[:space:]]*\\(qemu_.*\\);/_\\1/p', '@INPUT@'])
     plugin_ldflags = ['-Wl,-exported_symbols_list,plugins/qemu-plugins-ld64.symbols']
-  else
-    plugin_ldflags = ['-Xlinker', '--dynamic-list=' + (meson.project_source_root() / 'plugins/qemu-plugins.symbols')]
   endif
 endif
 
