From 4107f8813ec012682ba5a3fdfeb03fc597c33f10 Mon Sep 17 00:00:00 2001
From: Dominik Hassler <hadfl@omnios.org>
Date: Sat, 10 May 2025 19:52:04 +0000
Subject: include sys/loadavg.h for getloadavg

---
 qga/commands-posix.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/qga/commands-posix.c b/qga/commands-posix.c
index 837be51c40..19cd12e854 100644
--- a/qga/commands-posix.c
+++ b/qga/commands-posix.c
@@ -50,6 +50,10 @@
 #endif
 #endif
 
+#ifdef __illumos__
+#include <sys/loadavg.h>
+#endif
+
 static bool ga_wait_child(pid_t pid, int *status, Error **errp)
 {
     pid_t rpid;
