illumos defines FSCALE in /usr/include/sys/param.h

diff -wpruN '--exclude=*.orig' a~/target/arm/translate-sve.c a/target/arm/translate-sve.c
--- a~/target/arm/translate-sve.c	1970-01-01 00:00:00
+++ a/target/arm/translate-sve.c	1970-01-01 00:00:00
@@ -3996,6 +3996,7 @@ DO_ZPZZ_FP(FMAX_zpzz, aa64_sve, sve_fmax
 DO_ZPZZ_FP(FMINNM_zpzz, aa64_sve, sve_fminnum)
 DO_ZPZZ_FP(FMAXNM_zpzz, aa64_sve, sve_fmaxnum)
 DO_ZPZZ_FP(FABD, aa64_sve, sve_fabd)
+#undef FSCALE
 DO_ZPZZ_FP(FSCALE, aa64_sve, sve_fscalbn)
 DO_ZPZZ_FP(FDIV, aa64_sve, sve_fdiv)
 DO_ZPZZ_FP(FMULX, aa64_sve, sve_fmulx)
