diff -wpruN --no-dereference '--exclude=*.orig' a~/CMakeLists.txt a/CMakeLists.txt
--- a~/CMakeLists.txt	1970-01-01 00:00:00
+++ a/CMakeLists.txt	1970-01-01 00:00:00
@@ -934,6 +934,10 @@ set(ZIG_BUILD_ARGS
   "-Dversion-string=${RESOLVED_ZIG_VERSION}"
 )
 
+if(ZIG_BUILD_VERBOSE_LINK)
+  list(APPEND ZIG_BUILD_ARGS "--verbose-link")
+endif()
+
 add_custom_target(stage3 ALL
   DEPENDS "${CMAKE_BINARY_DIR}/stage3/bin/zig"
 )
