diff -wpruN --no-dereference '--exclude=*.orig' a~/ui/cmdpipe.c a/ui/cmdpipe.c
--- a~/ui/cmdpipe.c	1970-01-01 00:00:00
+++ a/ui/cmdpipe.c	1970-01-01 00:00:00
@@ -230,7 +230,7 @@ void execute_packet_child(
     if (access ("/etc/mtr.is.run.under.sudo", F_OK) != 0)
         mtr_packet_path = getenv("MTR_PACKET");
     if (mtr_packet_path == NULL)
-        mtr_packet_path = "mtr-packet";
+        mtr_packet_path = MTR_PACKET_PATH;
 
     /*
        First, try to execute mtr-packet from PATH
