diff -wpruN --no-dereference '--exclude=*.orig' a~/Makefile.am a/Makefile.am
--- a~/Makefile.am	1970-01-01 00:00:00
+++ a/Makefile.am	1970-01-01 00:00:00
@@ -39,7 +39,7 @@ man_MANS = $(interdiff_manpage) $(other_
 
 $(other_manpages): $(interdiff_manpage)
 $(interdiff_manpage): doc/patchutils.xml
-	$(XMLTO) man -o doc $<
+	$(XMLTO) --skip-validation man -o doc $<
 endif
 
 # Special rules for combinediff, flipdiff, lsdiff, grepdiff and patchview, which are
