diff -wpruN '--exclude=*.orig' a~/scheduler/cups.xml.in a/scheduler/cups.xml.in
--- a~/scheduler/cups.xml.in	1970-01-01 00:00:00
+++ a/scheduler/cups.xml.in	1970-01-01 00:00:00
@@ -49,14 +49,6 @@
         <service_fmri value='svc:/milestone/network' />
     </dependency>
 
-    <dependency
-        name='identity-cert'
-        grouping='require_all'
-        restart_on='refresh'
-        type='service'>
-        <service_fmri value='svc:/system/identity:cert'/>
-    </dependency>
-
     <exec_method
         type='method'
         name='stop'
@@ -78,7 +70,7 @@
             name='start'
             exec='@sbindir@/cupsd'
             timeout_seconds='60' >
-            <method_context>
+            <method_context security_flags="aslr">
                 <method_credential user='root' group='@CUPS_GROUP@' />
             </method_context>
         </exec_method>
@@ -165,7 +157,7 @@
             name='inetd_start'
             exec='@CUPS_SERVERBIN@/daemon/cups-lpd -o document-format=application/octet-stream'
             timeout_seconds='0'>
-            <method_context>
+            <method_context security_flags="aslr">
                 <method_credential user='@CUPS_USER@' group='@CUPS_GROUP@' />
             </method_context>
         </exec_method>
