diff -wpruN --no-dereference '--exclude=*.orig' a~/config-scripts/cups-compiler.m4 a/config-scripts/cups-compiler.m4
--- a~/config-scripts/cups-compiler.m4	1970-01-01 00:00:00
+++ a/config-scripts/cups-compiler.m4	1970-01-01 00:00:00
@@ -164,7 +164,7 @@ AS_IF([test -n "$GCC"], [
 	    ])
 	], [*], [
 	    CFLAGS="$CFLAGS -fPIE -pie"
-	    AC_COMPILE_IFELSE([AC_LANG_PROGRAM()], [
+	    AC_LINK_IFELSE([AC_LANG_PROGRAM()], [
 		PIEFLAGS="-fPIE -pie"
 		AC_MSG_RESULT([yes])
 	    ], [
